
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Temperature, scaling, enzyme kinetics, and exponential growth rate &#8212; MulQuaBio</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=3943d584" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css?v=01a0bdc2" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/togglebutton.js?v=3e7854f7"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notebooks/MetabolicBasis_additonal';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/logo.png" class="logo__image only-light" alt="MulQuaBio - Home"/>
    <script>document.write(`<img src="../_static/logo.png" class="logo__image only-dark" alt="MulQuaBio - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Welcome to The Multilingual Quantitative Biologist!
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Computing</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Unix.html">UNIX and Linux</a></li>
<li class="toctree-l1"><a class="reference internal" href="ShellScripting.html">Shell scripting</a></li>
<li class="toctree-l1"><a class="reference internal" href="Git.html">Version control with Git</a></li>
<li class="toctree-l1"><a class="reference internal" href="LaTeX.html">Scientific documents with <span class="math notranslate nohighlight">\(\LaTeX\)</span></a></li>
<li class="toctree-l1"><a class="reference internal" href="Python.html">Biological Computing in Python</a></li>
<li class="toctree-l1"><a class="reference internal" href="R.html">Biological Computing in R</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Ecological &amp; Evolutionary Modelling</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="MetabolicBasis.html">The Metabolic Basis of Ecology and Evolutionary dynamics</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Basic Data Analyses and Statistics</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../intro-Stats.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="Data_R.html">Data Management and Visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="ExpDesign.html">Experimental design and Data exploration</a></li>
<li class="toctree-l1"><a class="reference internal" href="t_F_tests.html">Basic hypothesis testing: <span class="math notranslate nohighlight">\(t\)</span> and <span class="math notranslate nohighlight">\(F\)</span> tests</a></li>
<li class="toctree-l1"><a class="reference internal" href="regress.html">Linear Models: Regression</a></li>
<li class="toctree-l1"><a class="reference internal" href="anova.html">Linear Models: Analysis of variance</a></li>
<li class="toctree-l1"><a class="reference internal" href="MulExpl.html">Linear Models: Multiple explanatory variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="MulExplInter.html">Linear Models: Multiple variables with interactions</a></li>
<li class="toctree-l1"><a class="reference internal" href="ModelSimp.html">Model simplification</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Advanced Data Analyses and Statistics</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="glm.html">Generalised Linear Models</a></li>
<li class="toctree-l1"><a class="reference internal" href="NLLS.html">Model Fitting using Non-linear Least-squares</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendices</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Appendix-JupyIntro.html">Introduction to Jupyter</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-Data-Python.html">Data analyses  with Python &amp; Jupyter</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-Maths.html">Mathematical models in Jupyter</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-MathsForBiologists/MathsForBiologists.html">Maths for Biologists</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-Databases.html">Databases <span class="tocSkip"></span></a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-NLLS-Python.html">Model fitting in Python</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-MiniProj.html">The Computing Miniproject</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-Assessment.html">TMQB Coursework Assessment</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-HighPerformanceComputing.html">Introduction to High-Performance Computing (HPC)</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/MulQuaBio/MQB" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/MulQuaBio/MQB/issues/new?title=Issue%20on%20page%20%2Fnotebooks/MetabolicBasis_additonal.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/notebooks/MetabolicBasis_additonal.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Temperature, scaling, enzyme kinetics, and exponential growth rate</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#introduction-and-historical-context">1. Introduction and Historical Context</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#arrhenius-equation">1.1. Arrhenius Equation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#eyring-equation-transition-state-theory">1.2. Eyring Equation (Transition State Theory)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#application-to-enzyme-kinetics">1.3. Application to Enzyme Kinetics</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#constants-and-helper-functions">2. Constants and Helper Functions</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#interactive-plots">3. Interactive Plots</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#discussion">4. Discussion</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#conclusion">5. Conclusion</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#enzymes-and-the-temperature-dependence-of-reaction-rates">1. Enzymes and the Temperature Dependence of Reaction Rates</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#arrhenius-or-eyringstyle-dependence">1.1 Arrhenius or Eyring‐Style Dependence</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#summation-of-intracellular-reaction-rates">1.2 Summation of Intracellular Reaction Rates</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#from-singlecell-growth-to-division">2. From Single‐Cell Growth to Division</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-division-as-a-stochastic-process">2.1 Cell Division as a (Stochastic) Process</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#average-doubling-time">2.2 Average Doubling Time</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#exponential-population-growth">3. Exponential Population Growth</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#linking-cell-division-rate-to-population-growth">3.1 Linking Cell Division Rate to Population Growth</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#growth-rate-vs-doubling-time">3.2 Growth Rate vs Doubling Time</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#putting-it-all-together">4. Putting It All Together</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#caveats-and-extensions">5. Caveats and Extensions</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#final-schematic-summary">6. Final Schematic Summary</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#temperaturedependent-enzyme-kinetics">1. Temperature‐Dependent Enzyme Kinetics</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">1.1 Arrhenius or Eyring‐Style Dependence</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-division-and-doubling-rate">2. Cell Division and Doubling Rate</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#division-as-a-stochastic-process">2.1 Division as a (Stochastic) Process</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">3. Exponential Population Growth</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#linking-division-rate-to-population-size">3.1 Linking Division Rate to Population Size</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#incorporating-cell-size-scaling-from-metabolic-theory-of-ecology-mte">4. Incorporating Cell Size Scaling from Metabolic Theory of Ecology (MTE)</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#core-premise-of-mte">4.1 Core Premise of MTE</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#growth-rate-from-metabolic-rate">4.2 Growth Rate from Metabolic Rate</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-size-scaling-factor">4.3 Cell Size Scaling Factor</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#exponential-growth-with-mte-scaling">4.4 Exponential Growth with MTE Scaling</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#id3">5. Putting It All Together</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#key-takeaways">6. Key Takeaways</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#recap-temperature-and-cell-size-dependence-in-exponential-growth">1. Recap: Temperature and Cell Size Dependence in Exponential Growth</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#logistic-growth-the-basic-model">2. Logistic Growth: The Basic Model</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#classic-logistic-equation">2.1 Classic Logistic Equation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#temperature-size-dependence-in-logistic-growth">2.2 Temperature &amp; Size Dependence in Logistic Growth</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#how-it-all-ties-together">3. How It All Ties Together</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#initial-lowdensity-phase-exponential-growth">3.1 Initial (Low‐Density) Phase: Exponential Growth</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#late-highdensity-phase-saturation-at-k">3.2 Late (High‐Density) Phase: Saturation at <span class="math notranslate nohighlight">\(K\)</span></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#possible-temperature-size-dependence-of-k">3.3 Possible Temperature/Size Dependence of <span class="math notranslate nohighlight">\(K\)</span></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#illustrative-equations">4. Illustrative Equations</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#biological-interpretation">5. Biological Interpretation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#final-summary">6. Final Summary</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#references">References</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#temperature-dependence-of-reaction-rates">1. Temperature Dependence of Reaction Rates</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#classical-microbial-growth-kinetics">2. Classical Microbial Growth Kinetics</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#logistic-growth-population-ecology">3. Logistic Growth &amp; Population Ecology</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#metabolic-theory-of-ecology-mte">4. Metabolic Theory of Ecology (MTE)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#integrative-synthetic-works">5. Integrative / Synthetic Works</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#additional-notes">Additional Notes:</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="temperature-scaling-enzyme-kinetics-and-exponential-growth-rate">
<h1>Temperature, scaling, enzyme kinetics, and exponential growth rate<a class="headerlink" href="#temperature-scaling-enzyme-kinetics-and-exponential-growth-rate" title="Link to this heading">#</a></h1>
<section id="introduction-and-historical-context">
<h2>1. Introduction and Historical Context<a class="headerlink" href="#introduction-and-historical-context" title="Link to this heading">#</a></h2>
<section id="arrhenius-equation">
<h3>1.1. Arrhenius Equation<a class="headerlink" href="#arrhenius-equation" title="Link to this heading">#</a></h3>
<p>One of the earliest attempts to capture the temperature dependence of reaction rates was pioneered by <strong>Jacobus Henricus van ’t Hoff</strong> (1852–1911), who observed an exponential increase in the rate of chemical reactions with increasing temperature. Later, <strong>Svante Arrhenius</strong> (1859–1927) formalized this relationship in the well-known <strong>Arrhenius Equation</strong>:</p>
<div class="math notranslate nohighlight">
\[\displaystyle k = A \exp\Bigl( -\frac{E_a}{RT} \Bigr),\]</div>
<p>where</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(k\)</span> is the rate constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(A\)</span> is the pre-exponential (frequency) factor,</p></li>
<li><p><span class="math notranslate nohighlight">\(E_a\)</span> is the activation energy,</p></li>
<li><p><span class="math notranslate nohighlight">\(R\)</span> is the universal gas constant, and</p></li>
<li><p><span class="math notranslate nohighlight">\(T\)</span> is the absolute temperature (in Kelvin).</p></li>
</ul>
<p>For <strong>enzyme-catalyzed reactions</strong>, the same principle applies for the <strong>catalytic rate constant</strong> <span class="math notranslate nohighlight">\(k_{\mathrm{cat}}\)</span>, but enzymes also have a <strong>thermal denaturation</strong> process at higher temperatures, causing activity to drop above a certain threshold.</p>
</section>
<section id="eyring-equation-transition-state-theory">
<h3>1.2. Eyring Equation (Transition State Theory)<a class="headerlink" href="#eyring-equation-transition-state-theory" title="Link to this heading">#</a></h3>
<p>In the 1930s, <strong>Henry Eyring</strong> (1901–1981) and others developed the <strong>Transition State Theory (TST)</strong>. The key result of TST is the <strong>Eyring-Polanyi Equation</strong>, usually written in the form:</p>
<div class="math notranslate nohighlight">
\[\displaystyle k = \kappa \frac{k_B \, T}{h} \exp\Bigl( \frac{\Delta S^\ddagger}{R}\Bigr) \exp\Bigl( - \frac{\Delta H^\ddagger}{RT} \Bigr),\]</div>
<p>where</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(\kappa\)</span> is the transmission coefficient (often assumed to be ~1 for many simple reactions),</p></li>
<li><p><span class="math notranslate nohighlight">\(k_B\)</span> is the Boltzmann constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(h\)</span> is Planck’s constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> is the entropy of activation,</p></li>
<li><p><span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> is the enthalpy of activation, and</p></li>
<li><p><span class="math notranslate nohighlight">\(R\)</span> and <span class="math notranslate nohighlight">\(T\)</span> are as before.</p></li>
</ul>
<p>This equation is sometimes called the “<strong>Eyring equation</strong>” and offers a thermodynamics-based perspective. It reduces to an Arrhenius-like form when you group constants appropriately.</p>
</section>
<section id="application-to-enzyme-kinetics">
<h3>1.3. Application to Enzyme Kinetics<a class="headerlink" href="#application-to-enzyme-kinetics" title="Link to this heading">#</a></h3>
<p>For an <strong>enzyme-catalyzed reaction</strong>, the rate can be written as:</p>
<div class="math notranslate nohighlight">
\[\displaystyle \text{Rate} = k_\text{cat}(T) \times [\text{E}] \times f([\text{S}]),\]</div>
<p>where:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(k_\text{cat}(T)\)</span> is the turnover number, strongly dependent on <span class="math notranslate nohighlight">\(T\)</span>,</p></li>
<li><p><span class="math notranslate nohighlight">\([\text{E}]\)</span> is the enzyme concentration,</p></li>
<li><p><span class="math notranslate nohighlight">\(f([\text{S}])\)</span> is a function of substrate concentration (such as the Michaelis-Menten form <span class="math notranslate nohighlight">\(\tfrac{[\text{S}]}{K_M + [\text{S}] }\)</span>).</p></li>
</ul>
<p>We can model <span class="math notranslate nohighlight">\(k_\text{cat}(T)\)</span> using either the <strong>Arrhenius Equation</strong> or the <strong>Eyring Equation</strong>. However, because real enzymes <strong>denature</strong> at higher temperatures, we often observe a <strong>bell-shaped</strong> temperature activity profile.</p>
<p>Now, we will:</p>
<ol class="arabic simple">
<li><p>Demonstrate both the <strong>Arrhenius</strong> and <strong>Eyring</strong> forms for the temperature dependence of <span class="math notranslate nohighlight">\(k_\text{cat}\)</span>.</p></li>
<li><p>Include a simple logistic denaturation function to mimic the loss of enzyme activity at high temperature.</p></li>
<li><p>Visualize the resulting enzyme activity curves and explore parameter effects.</p></li>
</ol>
</section>
</section>
<section id="constants-and-helper-functions">
<h2>2. Constants and Helper Functions<a class="headerlink" href="#constants-and-helper-functions" title="Link to this heading">#</a></h2>
<p>We’ll define relevant physical constants and implement two functions for the temperature-dependent rate constants:</p>
<ol class="arabic simple">
<li><p><strong>Arrhenius Rate</strong>:<br />
$<span class="math notranslate nohighlight">\(\displaystyle k_{\text{Arrhenius}}(T) = A \exp\Bigl(-\frac{E_a}{RT}\Bigr).\)</span>$</p></li>
<li><p><strong>Eyring Rate</strong>:<br />
$<span class="math notranslate nohighlight">\(\displaystyle k_{\text{Eyring}}(T) = \kappa \frac{k_B T}{h} \exp\Bigl(\frac{\Delta S^\ddagger}{R}\Bigr) \exp\Bigl(-\frac{\Delta H^\ddagger}{RT}\Bigr).\)</span>$</p></li>
</ol>
<p>(Assume <span class="math notranslate nohighlight">\(\kappa \approx 1\)</span> for simplicity.)</p>
<p>In both cases, we’ll multiply by a logistic <strong>denaturation factor</strong> to represent the fraction of enzyme that remains active at high temperature.</p>
<p>Refer to <strong>panel (c)</strong> in the attached figure for how the Eyring equation can be fitted to data to extract thermodynamic parameters.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Physical constants</span>
<span class="n">R</span> <span class="o">=</span> <span class="mf">8.314</span>            <span class="c1"># J/(mol*K) - Universal gas constant</span>
<span class="n">k_B</span> <span class="o">=</span> <span class="mf">1.380649e-23</span>   <span class="c1"># J/K - Boltzmann constant</span>
<span class="n">h</span> <span class="o">=</span> <span class="mf">6.62607015e-34</span>   <span class="c1"># J*s - Planck&#39;s constant</span>

<span class="k">def</span> <span class="nf">arrhenius_rate</span><span class="p">(</span><span class="n">T_celsius</span><span class="p">,</span> <span class="n">A</span><span class="o">=</span><span class="mf">1e7</span><span class="p">,</span> <span class="n">Ea</span><span class="o">=</span><span class="mf">5e4</span><span class="p">,</span> <span class="n">T_denature</span><span class="o">=</span><span class="mf">60.0</span><span class="p">,</span> <span class="n">slope</span><span class="o">=</span><span class="mf">0.5</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Returns the enzyme-catalyzed reaction rate (arbitrary units)</span>
<span class="sd">    using the Arrhenius form combined with a logistic denaturation factor.</span>
<span class="sd">    </span>
<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    T_celsius : float</span>
<span class="sd">        Temperature in Celsius</span>
<span class="sd">    A : float</span>
<span class="sd">        Pre-exponential factor for Arrhenius</span>
<span class="sd">    Ea : float</span>
<span class="sd">        Activation energy in J/mol</span>
<span class="sd">    T_denature : float</span>
<span class="sd">        Approximate denaturation temperature in Celsius</span>
<span class="sd">    slope : float</span>
<span class="sd">        Controls the steepness of the logistic denaturation</span>
<span class="sd">    </span>
<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">    rate : float</span>
<span class="sd">        Modeled enzyme rate at the given temperature</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">T_kelvin</span> <span class="o">=</span> <span class="n">T_celsius</span> <span class="o">+</span> <span class="mf">273.15</span>
    
    <span class="c1"># Arrhenius part</span>
    <span class="n">k_cat</span> <span class="o">=</span> <span class="n">A</span> <span class="o">*</span> <span class="n">np</span><span class="o">.</span><span class="n">exp</span><span class="p">(</span><span class="o">-</span><span class="n">Ea</span> <span class="o">/</span> <span class="p">(</span><span class="n">R</span> <span class="o">*</span> <span class="n">T_kelvin</span><span class="p">))</span>
    
    <span class="c1"># Logistic denaturation factor</span>
    <span class="n">fraction_active</span> <span class="o">=</span> <span class="mf">1.0</span> <span class="o">/</span> <span class="p">(</span><span class="mf">1.0</span> <span class="o">+</span> <span class="n">np</span><span class="o">.</span><span class="n">exp</span><span class="p">(</span><span class="n">slope</span> <span class="o">*</span> <span class="p">(</span><span class="n">T_celsius</span> <span class="o">-</span> <span class="n">T_denature</span><span class="p">)))</span>
    
    <span class="k">return</span> <span class="n">k_cat</span> <span class="o">*</span> <span class="n">fraction_active</span>

<span class="k">def</span> <span class="nf">eyring_rate</span><span class="p">(</span><span class="n">T_celsius</span><span class="p">,</span> <span class="n">dH</span><span class="o">=</span><span class="mf">5e4</span><span class="p">,</span> <span class="n">dS</span><span class="o">=-</span><span class="mi">100</span><span class="p">,</span> <span class="n">T_denature</span><span class="o">=</span><span class="mf">60.0</span><span class="p">,</span> <span class="n">slope</span><span class="o">=</span><span class="mf">0.5</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Returns the enzyme-catalyzed reaction rate (arbitrary units)</span>
<span class="sd">    using the Eyring (transition state) form combined with a logistic denaturation factor.</span>
<span class="sd">    </span>
<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    T_celsius : float</span>
<span class="sd">        Temperature in Celsius</span>
<span class="sd">    dH : float</span>
<span class="sd">        Enthalpy of activation (J/mol)</span>
<span class="sd">    dS : float</span>
<span class="sd">        Entropy of activation (J/(mol*K))</span>
<span class="sd">    T_denature : float</span>
<span class="sd">        Approximate denaturation temperature in Celsius</span>
<span class="sd">    slope : float</span>
<span class="sd">        Controls the steepness of the logistic denaturation</span>
<span class="sd">    </span>
<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">    rate : float</span>
<span class="sd">        Modeled enzyme rate at the given temperature</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">T_kelvin</span> <span class="o">=</span> <span class="n">T_celsius</span> <span class="o">+</span> <span class="mf">273.15</span>
    
    <span class="c1"># Eyring part (assuming transmission coefficient ~ 1)</span>
    <span class="c1"># kB*T/h in s^-1, multiplied by exponent factors</span>
    <span class="n">k_cat</span> <span class="o">=</span> <span class="p">(</span><span class="n">k_B</span> <span class="o">*</span> <span class="n">T_kelvin</span> <span class="o">/</span> <span class="n">h</span><span class="p">)</span> <span class="o">*</span> <span class="n">np</span><span class="o">.</span><span class="n">exp</span><span class="p">(</span><span class="n">dS</span> <span class="o">/</span> <span class="n">R</span><span class="p">)</span> <span class="o">*</span> <span class="n">np</span><span class="o">.</span><span class="n">exp</span><span class="p">(</span><span class="o">-</span><span class="n">dH</span> <span class="o">/</span> <span class="p">(</span><span class="n">R</span> <span class="o">*</span> <span class="n">T_kelvin</span><span class="p">))</span>
    
    <span class="c1"># Logistic denaturation factor</span>
    <span class="n">fraction_active</span> <span class="o">=</span> <span class="mf">1.0</span> <span class="o">/</span> <span class="p">(</span><span class="mf">1.0</span> <span class="o">+</span> <span class="n">np</span><span class="o">.</span><span class="n">exp</span><span class="p">(</span><span class="n">slope</span> <span class="o">*</span> <span class="p">(</span><span class="n">T_celsius</span> <span class="o">-</span> <span class="n">T_denature</span><span class="p">)))</span>
    
    <span class="k">return</span> <span class="n">k_cat</span> <span class="o">*</span> <span class="n">fraction_active</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="interactive-plots">
<h2>3. Interactive Plots<a class="headerlink" href="#interactive-plots" title="Link to this heading">#</a></h2>
<p>Below, we create two interactive plots:</p>
<ol class="arabic simple">
<li><p><strong>Arrhenius-based</strong> temperature dependence.</p></li>
<li><p><strong>Eyring-based</strong> temperature dependence.</p></li>
</ol>
<p>In both cases, you can adjust parameters such as the activation energy/enthalpy (<span class="math notranslate nohighlight">\(E_a\)</span> or <span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span>), the entropy of activation (<span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span>), and the denaturation temperature or slope to see how the rate curve is affected.</p>
<p>Refer to <strong>panel (d)</strong> in the figure for how Michaelis–Menten kinetics specifically depends on <span class="math notranslate nohighlight">\(k_{\mathrm{cat}}\)</span> and <span class="math notranslate nohighlight">\(K_M\)</span> with temperature, but here we focus on the temperature dependence of <span class="math notranslate nohighlight">\(k_{\mathrm{cat}}\)</span> itself for simplicity.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">plot_arrhenius</span><span class="p">(</span><span class="n">A</span><span class="o">=</span><span class="mf">1e7</span><span class="p">,</span> <span class="n">Ea</span><span class="o">=</span><span class="mf">5e4</span><span class="p">,</span> <span class="n">T_denature</span><span class="o">=</span><span class="mf">60.0</span><span class="p">,</span> <span class="n">slope</span><span class="o">=</span><span class="mf">0.5</span><span class="p">):</span>
    <span class="n">T</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">100</span><span class="p">,</span> <span class="mi">200</span><span class="p">)</span>  <span class="c1"># Temperature in Celsius</span>
    <span class="n">rates</span> <span class="o">=</span> <span class="p">[</span><span class="n">arrhenius_rate</span><span class="p">(</span><span class="n">t</span><span class="p">,</span> <span class="n">A</span><span class="o">=</span><span class="n">A</span><span class="p">,</span> <span class="n">Ea</span><span class="o">=</span><span class="n">Ea</span><span class="p">,</span> <span class="n">T_denature</span><span class="o">=</span><span class="n">T_denature</span><span class="p">,</span> <span class="n">slope</span><span class="o">=</span><span class="n">slope</span><span class="p">)</span> <span class="k">for</span> <span class="n">t</span> <span class="ow">in</span> <span class="n">T</span><span class="p">]</span>
    
    <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">T</span><span class="p">,</span> <span class="n">rates</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Arrhenius-based Enzyme Rate&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">T_denature</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;r&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Denaturation Temp&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;Temperature (°C)&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;Rate (arbitrary units)&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Arrhenius + Denaturation Model&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="k">def</span> <span class="nf">plot_eyring</span><span class="p">(</span><span class="n">dH</span><span class="o">=</span><span class="mf">5e4</span><span class="p">,</span> <span class="n">dS</span><span class="o">=-</span><span class="mi">100</span><span class="p">,</span> <span class="n">T_denature</span><span class="o">=</span><span class="mf">60.0</span><span class="p">,</span> <span class="n">slope</span><span class="o">=</span><span class="mf">0.5</span><span class="p">):</span>
    <span class="n">T</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">100</span><span class="p">,</span> <span class="mi">200</span><span class="p">)</span>  <span class="c1"># Temperature in Celsius</span>
    <span class="n">rates</span> <span class="o">=</span> <span class="p">[</span><span class="n">eyring_rate</span><span class="p">(</span><span class="n">t</span><span class="p">,</span> <span class="n">dH</span><span class="o">=</span><span class="n">dH</span><span class="p">,</span> <span class="n">dS</span><span class="o">=</span><span class="n">dS</span><span class="p">,</span> <span class="n">T_denature</span><span class="o">=</span><span class="n">T_denature</span><span class="p">,</span> <span class="n">slope</span><span class="o">=</span><span class="n">slope</span><span class="p">)</span> <span class="k">for</span> <span class="n">t</span> <span class="ow">in</span> <span class="n">T</span><span class="p">]</span>
    
    <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span><span class="mi">5</span><span class="p">))</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">T</span><span class="p">,</span> <span class="n">rates</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Eyring-based Enzyme Rate&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">axvline</span><span class="p">(</span><span class="n">T_denature</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;r&#39;</span><span class="p">,</span> <span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Denaturation Temp&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s1">&#39;Temperature (°C)&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s1">&#39;Rate (arbitrary units)&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">title</span><span class="p">(</span><span class="s1">&#39;Eyring + Denaturation Model&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="n">interact</span><span class="p">(</span><span class="n">plot_arrhenius</span><span class="p">,</span>
         <span class="n">A</span><span class="o">=</span><span class="n">FloatSlider</span><span class="p">(</span><span class="n">value</span><span class="o">=</span><span class="mf">1e7</span><span class="p">,</span> <span class="nb">min</span><span class="o">=</span><span class="mf">1e5</span><span class="p">,</span> <span class="nb">max</span><span class="o">=</span><span class="mf">1e9</span><span class="p">,</span> <span class="n">step</span><span class="o">=</span><span class="mf">1e5</span><span class="p">,</span> <span class="n">continuous_update</span><span class="o">=</span><span class="kc">False</span><span class="p">),</span>
         <span class="n">Ea</span><span class="o">=</span><span class="n">FloatSlider</span><span class="p">(</span><span class="n">value</span><span class="o">=</span><span class="mf">5e4</span><span class="p">,</span> <span class="nb">min</span><span class="o">=</span><span class="mf">1e4</span><span class="p">,</span> <span class="nb">max</span><span class="o">=</span><span class="mf">2e5</span><span class="p">,</span> <span class="n">step</span><span class="o">=</span><span class="mf">1e4</span><span class="p">,</span> <span class="n">continuous_update</span><span class="o">=</span><span class="kc">False</span><span class="p">),</span>
         <span class="n">T_denature</span><span class="o">=</span><span class="n">FloatSlider</span><span class="p">(</span><span class="n">value</span><span class="o">=</span><span class="mi">60</span><span class="p">,</span> <span class="nb">min</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="nb">max</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span> <span class="n">step</span><span class="o">=</span><span class="mf">1.0</span><span class="p">,</span> <span class="n">continuous_update</span><span class="o">=</span><span class="kc">False</span><span class="p">),</span>
         <span class="n">slope</span><span class="o">=</span><span class="n">FloatSlider</span><span class="p">(</span><span class="n">value</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="nb">min</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span> <span class="nb">max</span><span class="o">=</span><span class="mf">2.0</span><span class="p">,</span> <span class="n">step</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span> <span class="n">continuous_update</span><span class="o">=</span><span class="kc">False</span><span class="p">));</span>

<span class="n">interact</span><span class="p">(</span><span class="n">plot_eyring</span><span class="p">,</span>
         <span class="n">dH</span><span class="o">=</span><span class="n">FloatSlider</span><span class="p">(</span><span class="n">value</span><span class="o">=</span><span class="mf">5e4</span><span class="p">,</span> <span class="nb">min</span><span class="o">=</span><span class="mf">1e4</span><span class="p">,</span> <span class="nb">max</span><span class="o">=</span><span class="mf">2e5</span><span class="p">,</span> <span class="n">step</span><span class="o">=</span><span class="mf">1e4</span><span class="p">,</span> <span class="n">continuous_update</span><span class="o">=</span><span class="kc">False</span><span class="p">),</span>
         <span class="n">dS</span><span class="o">=</span><span class="n">FloatSlider</span><span class="p">(</span><span class="n">value</span><span class="o">=-</span><span class="mi">100</span><span class="p">,</span> <span class="nb">min</span><span class="o">=-</span><span class="mi">300</span><span class="p">,</span> <span class="nb">max</span><span class="o">=</span><span class="mi">300</span><span class="p">,</span> <span class="n">step</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">continuous_update</span><span class="o">=</span><span class="kc">False</span><span class="p">),</span>
         <span class="n">T_denature</span><span class="o">=</span><span class="n">FloatSlider</span><span class="p">(</span><span class="n">value</span><span class="o">=</span><span class="mi">60</span><span class="p">,</span> <span class="nb">min</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="nb">max</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span> <span class="n">step</span><span class="o">=</span><span class="mf">1.0</span><span class="p">,</span> <span class="n">continuous_update</span><span class="o">=</span><span class="kc">False</span><span class="p">),</span>
         <span class="n">slope</span><span class="o">=</span><span class="n">FloatSlider</span><span class="p">(</span><span class="n">value</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="nb">min</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span> <span class="nb">max</span><span class="o">=</span><span class="mf">2.0</span><span class="p">,</span> <span class="n">step</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span> <span class="n">continuous_update</span><span class="o">=</span><span class="kc">False</span><span class="p">));</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<script type="application/vnd.jupyter.widget-view+json">{"model_id": "b896ba98ba334263b96e634ff0ae503e", "version_major": 2, "version_minor": 0}</script><script type="application/vnd.jupyter.widget-view+json">{"model_id": "ccefda973a474e53a9083d42c99f8e66", "version_major": 2, "version_minor": 0}</script></div>
</div>
</section>
<section id="discussion">
<h2>4. Discussion<a class="headerlink" href="#discussion" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>Arrhenius vs. Eyring</strong></p>
<ul class="simple">
<li><p>The <strong>Arrhenius Equation</strong> is an empirical formula that captures the exponential increase of reaction rate with temperature.</p></li>
<li><p>The <strong>Eyring Equation</strong> derives from <strong>Transition State Theory</strong>, relating rate constants to thermodynamic parameters of the transition state (<span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span>, <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span>), as depicted in panels (b) and (c) of the figure.</p></li>
</ul>
</li>
<li><p><strong>Role of Denaturation</strong></p>
<ul class="simple">
<li><p>Both models in this notebook include a <em>logistic</em> denaturation factor to reduce active enzyme at high temperatures. Real systems can have more intricate denaturation kinetics, but this simple approach captures the key idea: above a certain threshold, enzyme function declines rapidly.</p></li>
</ul>
</li>
<li><p><strong>Bell-Shaped Curve</strong></p>
<ul class="simple">
<li><p>Combining an <strong>Arrhenius- or Eyring-like increase</strong> at moderate temperatures with <strong>denaturation</strong> at higher temperatures typically yields the <strong>bell-shaped</strong> temperature-activity curve depicted in many enzyme-kinetics references.</p></li>
<li><p>As shown in panel (d), enzymatic rates may also be viewed in the context of substrate concentration and Michaelis–Menten parameters, but the main driver at high temperature is denaturation.</p></li>
</ul>
</li>
<li><p><strong>Parameter Interpretation</strong></p>
<ul class="simple">
<li><p><strong>Arrhenius</strong>: <span class="math notranslate nohighlight">\(E_a\)</span> (J/mol) is the activation energy barrier.</p></li>
<li><p><strong>Eyring</strong>: <span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> (J/mol) is the enthalpy of activation; <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> (J/(mol·K)) is the entropy of activation. A large positive <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> can lead to higher rates, while a large positive <span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> lowers the rate.</p></li>
</ul>
</li>
<li><p><strong>Historical Significance</strong></p>
<ul class="simple">
<li><p><strong>Arrhenius</strong> (1889) built on <strong>van ’t Hoff’s</strong> work on temperature dependence of equilibrium constants and reaction rates.</p></li>
<li><p><strong>Henry Eyring</strong> (1935) developed Transition State Theory, leading to the <strong>Eyring-Polanyi equation</strong>, providing a more fundamental theoretical underpinning for temperature-dependent rate processes in chemistry and biochemistry.</p></li>
</ul>
</li>
</ol>
</section>
<section id="conclusion">
<h2>5. Conclusion<a class="headerlink" href="#conclusion" title="Link to this heading">#</a></h2>
<p>In this notebook, we have:</p>
<ul class="simple">
<li><p>Incorporated insights from a <strong>recent review figure</strong> (panels (a)–(e)) showing Arrhenius, Eyring, and Michaelis–Menten perspectives.</p></li>
<li><p>Shown how <strong>Arrhenius</strong> and <strong>Eyring</strong> equations each describe temperature-dependent reaction rates.</p></li>
<li><p>Applied both to a simplified <strong>enzyme kinetics</strong> scenario, including a denaturation term.</p></li>
<li><p>Provided interactive sliders to explore how parameters like <span class="math notranslate nohighlight">\(E_a\)</span>, <span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span>, <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span>, and the denaturation threshold affect the enzyme’s activity profile.</p></li>
</ul>
<p>The combination of these models, especially with the logistic denaturation factor, nicely illustrates the <strong>bell-shaped</strong> temperature dependence typically observed in real enzymes. As the attached figure shows, such analyses can be extended to full <strong>Michaelis–Menten</strong> kinetics (panel d) and to distinguishing <strong>enthalpic</strong> vs. <strong>entropic</strong> components of catalysis (panel e).</p>
<p>Next, a conceptual <strong>derivation</strong> of how the <strong>temperature dependence</strong> of <strong>enzyme kinetics</strong> and basic <strong>cellular physiology</strong> leads to a <strong>temperature‐dependent</strong> rate of <strong>cell division</strong>, and ultimately to <strong>exponential population growth</strong> in microbes. The argument starts at the molecular level (enzyme‐catalyzed reactions) and progresses up to the population level (exponential growth).</p>
</section>
<hr class="docutils" />
<section id="enzymes-and-the-temperature-dependence-of-reaction-rates">
<h2>1. Enzymes and the Temperature Dependence of Reaction Rates<a class="headerlink" href="#enzymes-and-the-temperature-dependence-of-reaction-rates" title="Link to this heading">#</a></h2>
<section id="arrhenius-or-eyringstyle-dependence">
<h3>1.1 Arrhenius or Eyring‐Style Dependence<a class="headerlink" href="#arrhenius-or-eyringstyle-dependence" title="Link to this heading">#</a></h3>
<p>At the <strong>molecular</strong> level, many key intracellular reactions follow temperature‐dependent kinetics that often can be approximated (over a modest temperature range) by an <strong>Arrhenius‐type</strong> expression:</p>
<div class="math notranslate nohighlight">
\[
k(T)  =  A\,\exp\!\Bigl(-\frac{E_a}{R\,T}\Bigr),
\]</div>
<p>where</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(k(T)\)</span> is the reaction rate constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(E_a\)</span> is the (apparent) activation energy,</p></li>
<li><p><span class="math notranslate nohighlight">\(A\)</span> is a pre‐exponential factor,</p></li>
<li><p><span class="math notranslate nohighlight">\(R\)</span> is the gas constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(T\)</span> is the absolute temperature.</p></li>
</ul>
<p>Alternatively, if one prefers a <strong>transition‐state theory</strong> perspective, the <strong>Eyring equation</strong> can be used:</p>
<div class="math notranslate nohighlight">
\[
k(T)  =  \frac{k_B T}{h} \,\exp\!\Bigl(\tfrac{\Delta S^\ddagger}{R}\Bigr)\,\exp\!\Bigl(-\tfrac{\Delta H^\ddagger}{R\,T}\Bigr),
\]</div>
<p>where <span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> and <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> are the activation enthalpy and entropy, respectively, and <span class="math notranslate nohighlight">\(k_B\)</span> and <span class="math notranslate nohighlight">\(h\)</span> are Boltzmann’s and Planck’s constants. The <strong>key takeaway</strong> is that <strong>reaction rates</strong> within the cell <strong>increase with temperature</strong> (until a certain thermal limit).</p>
</section>
<section id="summation-of-intracellular-reaction-rates">
<h3>1.2 Summation of Intracellular Reaction Rates<a class="headerlink" href="#summation-of-intracellular-reaction-rates" title="Link to this heading">#</a></h3>
<p>A living cell’s <strong>growth and division</strong> require a <strong>multitude of coupled enzyme reactions</strong> (for nutrient uptake, biosynthesis, energy generation, etc.). In many simplified models, <strong>one key rate‐limiting step</strong> (or a small set of them) dominates the overall pace of cell growth and division. If those rate‐limiting reactions follow Arrhenius‐type kinetics, then the <strong>cellular physiological rate</strong> (e.g., biomass production rate, cell cycle progression rate) will inherit a similar temperature dependence:</p>
<div class="math notranslate nohighlight">
\[
\text{Physiological rate}  \propto  k(T)  \approx  A\,\exp\!\Bigl(-\tfrac{E_a}{R\,T}\Bigr).
\]</div>
<p>Hence, as <span class="math notranslate nohighlight">\(T\)</span> increases (within a tolerable range), <strong>enzyme‐catalyzed processes speed up</strong>, and so does the <strong>cell’s overall metabolism</strong> and ability to replicate.</p>
</section>
</section>
<hr class="docutils" />
<section id="from-singlecell-growth-to-division">
<h2>2. From Single‐Cell Growth to Division<a class="headerlink" href="#from-singlecell-growth-to-division" title="Link to this heading">#</a></h2>
<section id="cell-division-as-a-stochastic-process">
<h3>2.1 Cell Division as a (Stochastic) Process<a class="headerlink" href="#cell-division-as-a-stochastic-process" title="Link to this heading">#</a></h3>
<p>To move from enzyme kinetics to <strong>cell‐level</strong> behavior, note that a <strong>single cell</strong> must:</p>
<ol class="arabic simple">
<li><p><strong>Replicate all essential components</strong> (DNA, proteins, membranes, etc.).</p></li>
<li><p>Reach a certain “completion threshold” in biomass or chromosome replication.</p></li>
<li><p><strong>Divide</strong> into two daughter cells.</p></li>
</ol>
<p>Because cellular biochemical events occur stochastically, the <em>exact</em> division time of any one cell can vary. However, in an <strong>exponential (log) growth phase</strong> with abundant nutrients, the <strong>average</strong> cell‐division rate <span class="math notranslate nohighlight">\(\lambda\)</span> (divisions per unit time) is fairly constant. This average division rate is ultimately <strong>tied</strong> to the cumulative rate of intracellular biochemical and biosynthetic reactions, and thus inherits their temperature dependence.</p>
</section>
<section id="average-doubling-time">
<h3>2.2 Average Doubling Time<a class="headerlink" href="#average-doubling-time" title="Link to this heading">#</a></h3>
<p>If <span class="math notranslate nohighlight">\(\tau_d\)</span> is the <strong>average doubling time</strong> (the mean time for one cell to become two), then</p>
<div class="math notranslate nohighlight">
\[
\lambda  =  \frac{1}{\tau_d}
  \Longleftrightarrow  
\tau_d  =  \frac{1}{\lambda}.
\]</div>
<p>From the <strong>Arrhenius perspective</strong>, we might write</p>
<div class="math notranslate nohighlight">
\[
\tau_d(T)  \approx  \tau_{d,0}\,\exp\!\Bigl(\tfrac{E_a}{R\,T}\Bigr),
\]</div>
<p>so that the cell’s doubling time <strong>decreases</strong> with <strong>increasing</strong> temperature (again, within the physiological limit). Equivalently, the <strong>division rate</strong> <span class="math notranslate nohighlight">\(\lambda(T) = 1/\tau_d(T)\)</span> would <strong>increase</strong> with temperature:</p>
<div class="math notranslate nohighlight">
\[
\lambda(T)  \approx  \lambda_0 \,\exp\!\Bigl(-\tfrac{E_a}{R\,T}\Bigr).
\]</div>
</section>
</section>
<hr class="docutils" />
<section id="exponential-population-growth">
<h2>3. Exponential Population Growth<a class="headerlink" href="#exponential-population-growth" title="Link to this heading">#</a></h2>
<section id="linking-cell-division-rate-to-population-growth">
<h3>3.1 Linking Cell Division Rate to Population Growth<a class="headerlink" href="#linking-cell-division-rate-to-population-growth" title="Link to this heading">#</a></h3>
<p>Now consider a well‐mixed microbial culture (in batch or chemostat conditions, but focusing on <strong>log phase</strong>). If each cell divides (on average) at rate <span class="math notranslate nohighlight">\(\lambda(T)\)</span>, then the <strong>expected</strong> number of cells <span class="math notranslate nohighlight">\(N(t)\)</span> follows the <strong>exponential growth law</strong>:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN(t)}{dt}  =  \lambda(T)\,N(t),
\]</div>
<div class="math notranslate nohighlight">
\[
\implies\quad N(t)  =  N(0)\,\exp\!\bigl[\lambda(T)\,t\bigr].
\]</div>
<p>This is simply the standard Malthus‐type (or exponential) growth equation, whose solution is an exponential in time, with the <strong>growth rate</strong> <span class="math notranslate nohighlight">\(\lambda(T)\)</span> now an <strong>Arrhenius‐like function</strong> of temperature.</p>
</section>
<section id="growth-rate-vs-doubling-time">
<h3>3.2 Growth Rate vs Doubling Time<a class="headerlink" href="#growth-rate-vs-doubling-time" title="Link to this heading">#</a></h3>
<p>It is common in microbiology to define the <strong>specific growth rate</strong> <span class="math notranslate nohighlight">\(\mu(T)\)</span> via</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}  =  \mu(T)\,N,
\]</div>
<p>so that <span class="math notranslate nohighlight">\(\mu(T)\equiv \lambda(T)\)</span>. Equivalently, the <strong>doubling time</strong> is</p>
<div class="math notranslate nohighlight">
\[
t_d  =  \frac{\ln(2)}{\mu(T)}.
\]</div>
<p>Thus, you will often see microbial growth curves written as</p>
<div class="math notranslate nohighlight">
\[
N(t)  =  N(0)\,\exp\!\bigl[\mu(T)\,t\bigr],
\]</div>
<p>where</p>
<div class="math notranslate nohighlight">
\[
\mu(T)  \approx  \mu_0\,\exp\!\Bigl(-\tfrac{E_a}{R\,T}\Bigr).
\]</div>
</section>
</section>
<hr class="docutils" />
<section id="putting-it-all-together">
<h2>4. Putting It All Together<a class="headerlink" href="#putting-it-all-together" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>Intracellular Enzyme Kinetics</strong><br />
<span class="math notranslate nohighlight">\( \to\)</span> Reaction rates speed up with temperature according to Arrhenius or Eyring equations.</p></li>
<li><p><strong>Cell‐Level Physiology</strong><br />
<span class="math notranslate nohighlight">\( \to\)</span> The <strong>cell division cycle</strong> (DNA replication, protein synthesis, etc.) is governed by these enzymatic rates, so the <strong>cell division rate</strong> <span class="math notranslate nohighlight">\(\lambda(T)\)</span> similarly increases with temperature (up to an optimum).</p></li>
<li><p><strong>Population Growth</strong><br />
<span class="math notranslate nohighlight">\( \to\)</span> When each cell divides at rate <span class="math notranslate nohighlight">\(\lambda(T)\)</span>, the <strong>population</strong> follows <strong>exponential growth</strong> <span class="math notranslate nohighlight">\(N(t) = N(0)\,\exp[\lambda(T)\,t]\)</span>. The <strong>specific growth rate</strong> <span class="math notranslate nohighlight">\(\mu(T)\)</span> is effectively the <strong>cell division rate</strong> and inherits the same temperature dependence.</p></li>
</ol>
</section>
<hr class="docutils" />
<section id="caveats-and-extensions">
<h2>5. Caveats and Extensions<a class="headerlink" href="#caveats-and-extensions" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>Optimal Temperature and Denaturation</strong></p>
<ul class="simple">
<li><p>Real microbes have an <strong>optimal growth temperature</strong>. At <strong>high temperatures</strong>, enzymes <strong>denature</strong>, and growth drops off. Thus, beyond a certain point, Arrhenius‐type behavior breaks down.</p></li>
</ul>
</li>
<li><p><strong>Complexity in Real Systems</strong></p>
<ul class="simple">
<li><p>Many enzymatic steps contribute to growth; the overall temperature dependence can be more complicated than a simple exponential. Empirical models like the <strong>Ratkowsky model</strong> (<span class="math notranslate nohighlight">\(\sqrt{\mu}\)</span> vs. <span class="math notranslate nohighlight">\(T\)</span>) or the <strong>Cardinal model</strong> with minimum, optimum, and maximum temperatures, are often used.</p></li>
</ul>
</li>
<li><p><strong>Maintenance Requirements</strong></p>
<ul class="simple">
<li><p>Below or near a certain threshold temperature, the <strong>maintenance energy</strong> requirement may exceed the energy‐generating reactions, effectively stopping growth.</p></li>
</ul>
</li>
<li><p><strong>Stochastic Division</strong></p>
<ul class="simple">
<li><p>In single‐cell studies (e.g., time‐lapse microscopy), you see that individual cells have <strong>variable</strong> interdivision times. However, at the <strong>population level</strong>, the <em>average</em> behavior typically follows the exponential law.</p></li>
</ul>
</li>
</ol>
</section>
<hr class="docutils" />
<section id="final-schematic-summary">
<h2>6. Final Schematic Summary<a class="headerlink" href="#final-schematic-summary" title="Link to this heading">#</a></h2>
<div class="math notranslate nohighlight">
\[\begin{split}
\begin{aligned}
&amp; \text{(1) Enzyme Kinetics:} \quad
   k \sim \exp\!\Bigl(-\tfrac{E_a}{R\,T}\Bigr) 
    \longrightarrow  \text{(2) Rate-Limiting Step in Cell Growth} \\
&amp; \quad \longrightarrow \text{Division Rate:}
     \lambda(T) \sim \exp\!\Bigl(-\tfrac{E_a}{R\,T}\Bigr) 
    \longrightarrow  \text{(3) Exponential Population Growth:} \\
&amp; \quad
   \frac{dN}{dt}  =  \lambda(T)\,N
   \quad\Longrightarrow\quad
   N(t)  =  N(0)\,\exp[\lambda(T)\,t].
\end{aligned}
\end{split}\]</div>
<p>Thus, <strong>starting</strong> from the <strong>temperature dependence</strong> of <strong>intracellular enzyme rates</strong>, we <strong>end</strong> with a <strong>temperature‐dependent exponential growth</strong> law at the <strong>population level</strong>.</p>
<p>Next, a <strong>conceptual derivation</strong> showing how <strong>temperature dependence</strong> in enzyme kinetics and <strong>cell‐level physiology</strong> leads to <strong>exponential microbial growth</strong>, but now <strong>including</strong> the <strong>cell size scaling factor</strong> from the <strong>Metabolic Theory of Ecology (MTE)</strong>. The argument spans:</p>
<ol class="arabic simple">
<li><p><strong>Intracellular enzyme kinetics</strong></p></li>
<li><p><strong>Cell division and doubling</strong></p></li>
<li><p><strong>Population‐level exponential growth</strong></p></li>
<li><p><strong>Scaling of metabolic/physiological rates with cell size (<span class="math notranslate nohighlight">\(M\)</span>)</strong> as formulated in MTE</p></li>
</ol>
</section>
<hr class="docutils" />
<section id="temperaturedependent-enzyme-kinetics">
<h2>1. Temperature‐Dependent Enzyme Kinetics<a class="headerlink" href="#temperaturedependent-enzyme-kinetics" title="Link to this heading">#</a></h2>
<section id="id1">
<h3>1.1 Arrhenius or Eyring‐Style Dependence<a class="headerlink" href="#id1" title="Link to this heading">#</a></h3>
<p>At the <strong>molecular level</strong>, many rate‐limiting steps (enzyme‐catalyzed reactions) show a temperature dependence that can often be approximated by an <strong>Arrhenius‐type</strong> expression:</p>
<div class="math notranslate nohighlight">
\[
k(T)  =  A\,\exp\!\Bigl(-\tfrac{E_a}{R\,T}\Bigr),
\]</div>
<p>or, from <strong>Transition State Theory</strong>, by the <strong>Eyring equation</strong>:</p>
<div class="math notranslate nohighlight">
\[
k(T)  =  \frac{k_B\, T}{h}\,\exp\!\Bigl(\tfrac{\Delta S^\ddagger}{R}\Bigr)\,\exp\!\Bigl(-\tfrac{\Delta H^\ddagger}{R\,T}\Bigr).
\]</div>
<p>In all cases, <strong>reaction rates tend to increase with temperature</strong> (until proteins denature at very high temperatures). A cell’s <strong>overall physiological rate</strong> (e.g., biomass synthesis, replication of DNA, etc.) is governed by the collective effect of many such reactions.</p>
</section>
</section>
<hr class="docutils" />
<section id="cell-division-and-doubling-rate">
<h2>2. Cell Division and Doubling Rate<a class="headerlink" href="#cell-division-and-doubling-rate" title="Link to this heading">#</a></h2>
<section id="division-as-a-stochastic-process">
<h3>2.1 Division as a (Stochastic) Process<a class="headerlink" href="#division-as-a-stochastic-process" title="Link to this heading">#</a></h3>
<p>A single microbial cell:</p>
<ol class="arabic simple">
<li><p>Grows (i.e., accumulates mass/biomass),</p></li>
<li><p>Replicates essential components (DNA, ribosomes, etc.),</p></li>
<li><p>Then <strong>divides</strong> once it reaches a sufficient size or completes the cell cycle.</p></li>
</ol>
<p>While <strong>stochastic</strong> at the single‐cell level, in <strong>log‐phase growth</strong>, the <strong>average</strong> cell‐division rate (cells per unit time) is fairly constant and is determined by the underlying biochemical (enzyme) rates. If we denote the <strong>mean cell‐division rate</strong> by <span class="math notranslate nohighlight">\(\lambda(T)\)</span>, it inherits a similar <strong>Arrhenius‐like</strong> temperature dependence:</p>
<div class="math notranslate nohighlight">
\[
\lambda(T)  \approx  \lambda_0\,\exp\!\Bigl(-\tfrac{E_a}{R\,T}\Bigr),
\]</div>
<p>within a biologically relevant temperature range. The <strong>average doubling time</strong> is then <span class="math notranslate nohighlight">\(\tau_d = 1/\lambda(T)\)</span>.</p>
</section>
</section>
<hr class="docutils" />
<section id="id2">
<h2>3. Exponential Population Growth<a class="headerlink" href="#id2" title="Link to this heading">#</a></h2>
<section id="linking-division-rate-to-population-size">
<h3>3.1 Linking Division Rate to Population Size<a class="headerlink" href="#linking-division-rate-to-population-size" title="Link to this heading">#</a></h3>
<p>For a well‐mixed microbial culture in <strong>exponential (log) growth</strong>:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}  =  \lambda(T)\,N(t),
\]</div>
<p>whose solution is
$<span class="math notranslate nohighlight">\(
N(t)  =  N(0)\,\exp\!\bigl[\lambda(T)\,t\bigr].
\)</span>$</p>
<p>Thus, the <strong>population</strong> grows <strong>exponentially</strong> at a <strong>temperature‐dependent</strong> rate <span class="math notranslate nohighlight">\(\lambda(T)\)</span>. Equivalently, one can define the <strong>specific growth rate</strong> <span class="math notranslate nohighlight">\(\mu(T) \equiv \lambda(T)\)</span>, so</p>
<div class="math notranslate nohighlight">
\[
N(t)  =  N(0)\,\exp\!\bigl[\mu(T)\,t\bigr], 
\quad
\mu(T)  \approx  \mu_0\,\exp\!\Bigl(-\tfrac{E_a}{R\,T}\Bigr).
\]</div>
</section>
</section>
<hr class="docutils" />
<section id="incorporating-cell-size-scaling-from-metabolic-theory-of-ecology-mte">
<h2>4. Incorporating Cell Size Scaling from Metabolic Theory of Ecology (MTE)<a class="headerlink" href="#incorporating-cell-size-scaling-from-metabolic-theory-of-ecology-mte" title="Link to this heading">#</a></h2>
<section id="core-premise-of-mte">
<h3>4.1 Core Premise of MTE<a class="headerlink" href="#core-premise-of-mte" title="Link to this heading">#</a></h3>
<p>The <strong>Metabolic Theory of Ecology</strong> posits that an organism’s <strong>mass‐specific metabolic rate</strong> depends on:</p>
<ol class="arabic simple">
<li><p><strong>Body (or cell) mass</strong> <span class="math notranslate nohighlight">\(M\)</span>, and</p></li>
<li><p><strong>Temperature</strong> <span class="math notranslate nohighlight">\(T\)</span>.</p></li>
</ol>
<p>In a classical MTE formulation, <strong>organismal metabolic rate</strong> <span class="math notranslate nohighlight">\(B\)</span> scales as:</p>
<div class="math notranslate nohighlight">
\[
B(M, T)  \approx  B_0\,M^{\,3/4}\,\exp\!\Bigl(-\tfrac{E}{k_B\,T}\Bigr),
\]</div>
<p>where:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(M\)</span> is the organism’s mass (for microbes, think <strong>cell mass</strong>),</p></li>
<li><p><span class="math notranslate nohighlight">\(E\)</span> is an <strong>effective activation energy</strong> for metabolic reactions,</p></li>
<li><p><span class="math notranslate nohighlight">\(k_B\)</span> here is Boltzmann’s constant (not to be confused with the same symbol in TST; context determines which constant is meant).</p></li>
</ul>
</section>
<section id="growth-rate-from-metabolic-rate">
<h3>4.2 Growth Rate from Metabolic Rate<a class="headerlink" href="#growth-rate-from-metabolic-rate" title="Link to this heading">#</a></h3>
<p>A common assumption in MTE is that the <strong>growth rate</strong> of an organism is roughly <strong>proportional</strong> to its <strong>mass‐specific</strong> metabolic rate:</p>
<div class="math notranslate nohighlight">
\[
\text{growth rate}  \propto  \frac{B(M,T)}{M}.
\]</div>
<p>Thus,</p>
<div class="math notranslate nohighlight">
\[
\text{growth rate}  \approx  \frac{B_0\,M^{3/4}\,\exp\!\bigl(-E/(k_B\,T)\bigr)}{M}
 = 
B_0\,M^{-1/4}\,\exp\!\bigl(-E/(k_B\,T)\bigr).
\]</div>
</section>
<section id="cell-size-scaling-factor">
<h3>4.3 Cell Size Scaling Factor<a class="headerlink" href="#cell-size-scaling-factor" title="Link to this heading">#</a></h3>
<p>For <strong>microbial cells</strong> of mass <span class="math notranslate nohighlight">\(M\)</span>, the MTE‐inspired prediction for the <strong>specific growth rate</strong> <span class="math notranslate nohighlight">\(\mu(T, M)\)</span> becomes:</p>
<div class="math notranslate nohighlight">
\[
\mu(T, M)
 \approx 
\mu_0 \,M^{-1/4}\,\exp\!\Bigl(-\tfrac{E}{k_B\,T}\Bigr),
\]</div>
<p>where <span class="math notranslate nohighlight">\(\mu_0\)</span> absorbs constants like <span class="math notranslate nohighlight">\(B_0\)</span>.</p>
<ul class="simple">
<li><p>The <strong><span class="math notranslate nohighlight">\(M^{-1/4}\)</span></strong> factor reflects how <strong>smaller cells</strong> typically have a <strong>faster</strong> mass‐specific metabolic rate (and thus can grow/divide faster), whereas larger microbial cells would have a somewhat lower per‐mass metabolic rate.</p></li>
</ul>
</section>
<section id="exponential-growth-with-mte-scaling">
<h3>4.4 Exponential Growth with MTE Scaling<a class="headerlink" href="#exponential-growth-with-mte-scaling" title="Link to this heading">#</a></h3>
<p>Plugging <span class="math notranslate nohighlight">\(\mu(T, M)\)</span> into the <strong>exponential growth equation</strong>:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}  =  \mu(T, M)\,N
\quad\Longrightarrow\quad
N(t)
 = 
N(0)\,\exp\!\Bigl[\mu(T, M)\,t\Bigr].
\]</div>
<p>Hence, from MTE, the <strong>population’s</strong> exponential growth rate now depends explicitly on <strong>both</strong>:</p>
<ol class="arabic simple">
<li><p><strong>Temperature</strong> (<span class="math notranslate nohighlight">\(\exp[-E/(k_B\,T)]\)</span> factor),</p></li>
<li><p><strong>Cell size</strong> (<span class="math notranslate nohighlight">\(M^{-1/4}\)</span> factor).</p></li>
</ol>
<p>In simpler microbial physiology models (which ignore size variation), one might set <span class="math notranslate nohighlight">\(M\approx\text{constant}\)</span>. But if comparing cells (or species) of <strong>different typical cell sizes</strong>, or tracking how cell size changes under various conditions, the <span class="math notranslate nohighlight">\(M^{-1/4}\)</span> dependence becomes relevant.</p>
</section>
</section>
<hr class="docutils" />
<section id="id3">
<h2>5. Putting It All Together<a class="headerlink" href="#id3" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>Enzymes and Temperature</strong></p>
<ul class="simple">
<li><p>Intracellular metabolic/biochemical <strong>rates</strong> follow an <strong>Arrhenius‐type</strong> or <strong>Eyring‐type</strong> increase with <span class="math notranslate nohighlight">\(T\)</span>.</p></li>
</ul>
</li>
<li><p><strong>Cell‐Level Growth and Division</strong></p>
<ul class="simple">
<li><p>The <strong>cell division</strong> (doubling) rate <span class="math notranslate nohighlight">\(\lambda(T)\)</span> (or <span class="math notranslate nohighlight">\(\mu(T)\)</span>) is governed by these enzyme‐limited processes.</p></li>
<li><p>In MTE, it also <strong>scales</strong> with <strong>cell mass</strong> as <span class="math notranslate nohighlight">\(M^{-1/4}\)</span>.</p></li>
</ul>
</li>
<li><p><strong>Population‐Level Exponential Growth</strong></p>
<ul class="simple">
<li><p>When every cell divides at rate <span class="math notranslate nohighlight">\(\mu(T,M)\)</span>, the population satisfies <span class="math notranslate nohighlight">\(\tfrac{dN}{dt} = \mu(T,M)\,N\)</span>.</p></li>
<li><p>Thus, we get <strong>exponential growth</strong> at a <strong>temperature‐ and size‐dependent</strong> rate.</p></li>
</ul>
</li>
</ol>
<div class="math notranslate nohighlight">
\[
\boxed{
N(t)  =  N(0)\,
\exp\!\Bigl[\mu_0\,M^{-1/4}\,\exp\!\bigl(-\tfrac{E}{k_B\,T}\bigr)\,t\Bigr].
}
\]</div>
</section>
<hr class="docutils" />
<section id="key-takeaways">
<h2>6. Key Takeaways<a class="headerlink" href="#key-takeaways" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p><strong>Classical enzyme kinetics</strong> (Arrhenius/Eyring) underlies the <strong>temperature dependence</strong> of cellular processes.</p></li>
<li><p><strong>Cellular physiology</strong> dictates that <strong>division rate</strong> scales with that enzymatic activity.</p></li>
<li><p><strong>Metabolic Theory of Ecology (MTE)</strong> adds the <strong>cell size (mass) scaling</strong>: metabolic rate <span class="math notranslate nohighlight">\(\sim M^{3/4}\)</span>, so <strong>growth rate</strong> <span class="math notranslate nohighlight">\(\sim M^{-1/4}\)</span>.</p></li>
<li><p>Combined, these yield a <strong>temperature‐ and size‐dependent</strong> exponential growth law:<br />
$<span class="math notranslate nohighlight">\(
  \mu(T,M) \sim M^{-1/4}\,\exp\!\Bigl(-\tfrac{E}{k_B\,T}\Bigr).
\)</span>$</p></li>
</ul>
<p>Thus, from the <strong>molecular</strong> level (enzyme kinetics) to the <strong>population</strong> level (exponential growth), incorporating the <strong>cell size</strong> factor from MTE unifies the <strong>thermal dependence</strong> of growth rates with <strong>allometric (size) scaling</strong>—providing a more holistic understanding of <strong>microbial growth</strong> across different <strong>temperatures</strong> and <strong>cell sizes</strong>.</p>
<p>Next a derivation and discussion of how <strong>temperature dependence</strong> (from enzyme kinetics and metabolic theory) plus <strong>cell size scaling</strong> can be integrated into a <strong>logistic growth</strong> framework. In other words, we move from the purely <strong>exponential</strong> growth regime (valid at low population density) to a <strong>density‐dependent</strong> (logistic) regime, where <strong>carrying capacity</strong> plays a role.</p>
</section>
<hr class="docutils" />
<section id="recap-temperature-and-cell-size-dependence-in-exponential-growth">
<h2>1. Recap: Temperature and Cell Size Dependence in Exponential Growth<a class="headerlink" href="#recap-temperature-and-cell-size-dependence-in-exponential-growth" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>Molecular Level (Arrhenius/Eyring)</strong><br />
$<span class="math notranslate nohighlight">\(
k(T)  \sim  \exp\!\bigl(-E_a/(R\,T)\bigr),
\)</span>$
capturing how intracellular (enzyme) reaction rates accelerate with increasing temperature (up to a limit).</p></li>
<li><p><strong>Cell‐Division Rate</strong><br />
$<span class="math notranslate nohighlight">\(
\mu(T,M)  \sim  M^{-1/4}\,\exp\!\Bigl(-\tfrac{E}{k_B\,T}\Bigr)
\)</span>$
(from <strong>Metabolic Theory of Ecology</strong>, MTE), where</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(M\)</span> is the cell mass,</p></li>
<li><p><span class="math notranslate nohighlight">\(E\)</span> is an effective activation energy for metabolism,</p></li>
<li><p>the factor <span class="math notranslate nohighlight">\(M^{-1/4}\)</span> encodes allometric scaling, so smaller cells often grow faster (per unit mass).</p></li>
</ul>
</li>
<li><p><strong>Exponential Growth</strong> (in unlimited resources)<br />
$<span class="math notranslate nohighlight">\(
\frac{dN}{dt}  =  \mu(T,M) N 
\quad\Longrightarrow\quad
N(t)  =  N(0)\,\exp\!\bigl[\mu(T,M)\,t\bigr].
\)</span>$</p></li>
</ol>
<p>However, in <strong>real</strong> microbial cultures, resources become depleted or waste products accumulate, preventing indefinite exponential growth. This leads to the <strong>logistic</strong> regime.</p>
</section>
<hr class="docutils" />
<section id="logistic-growth-the-basic-model">
<h2>2. Logistic Growth: The Basic Model<a class="headerlink" href="#logistic-growth-the-basic-model" title="Link to this heading">#</a></h2>
<section id="classic-logistic-equation">
<h3>2.1 Classic Logistic Equation<a class="headerlink" href="#classic-logistic-equation" title="Link to this heading">#</a></h3>
<p>The <strong>logistic equation</strong> introduces a <strong>carrying capacity</strong> <span class="math notranslate nohighlight">\(K\)</span>, which is the maximum population (or density) sustainable by the environment:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt} 
 =  r\,N\!\Bigl(1 - \frac{N}{K}\Bigr),
\]</div>
<p>where</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(r\)</span> is the (intrinsic) per‐capita growth rate in the <strong>absence</strong> of density‐dependent limitation,</p></li>
<li><p><span class="math notranslate nohighlight">\(K\)</span> is the <strong>carrying capacity</strong>.</p></li>
</ul>
<p><strong>Solution</strong> (if <span class="math notranslate nohighlight">\(r\)</span> and <span class="math notranslate nohighlight">\(K\)</span> are constants) is well known:</p>
<div class="math notranslate nohighlight">
\[
N(t) 
 =  
\frac{K}{\,1 + \bigl(\tfrac{K}{N(0)} - 1\bigr)\exp(-r\,t)\,},
\]</div>
<p>which saturates at <span class="math notranslate nohighlight">\(N(t)\to K\)</span> as <span class="math notranslate nohighlight">\(t\to\infty\)</span>.</p>
</section>
<section id="temperature-size-dependence-in-logistic-growth">
<h3>2.2 Temperature &amp; Size Dependence in Logistic Growth<a class="headerlink" href="#temperature-size-dependence-in-logistic-growth" title="Link to this heading">#</a></h3>
<p>To generalize, we can let the <strong>intrinsic growth rate</strong> <span class="math notranslate nohighlight">\(r\)</span> be <strong>temperature</strong>‐ and <strong>cell size</strong>‐dependent, i.e.,</p>
<div class="math notranslate nohighlight">
\[
r(T,M)
 \approx 
r_0\,M^{-1/4}\,\exp\!\Bigl(-\tfrac{E}{k_B\,T}\Bigr).
\]</div>
<p>Thus, the logistic equation becomes:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}
 = 
r(T,M)\,N  \Bigl(1 - \frac{N}{K}\Bigr).
\]</div>
<p>Depending on the complexity of the system, one might also consider <span class="math notranslate nohighlight">\(K\)</span> to be <strong>T‐dependent</strong>, <span class="math notranslate nohighlight">\(M\)</span>‐dependent, or both—especially if resource supply, waste removal, or cell size changes systematically with temperature or over time.</p>
</section>
</section>
<hr class="docutils" />
<section id="how-it-all-ties-together">
<h2>3. How It All Ties Together<a class="headerlink" href="#how-it-all-ties-together" title="Link to this heading">#</a></h2>
<section id="initial-lowdensity-phase-exponential-growth">
<h3>3.1 Initial (Low‐Density) Phase: Exponential Growth<a class="headerlink" href="#initial-lowdensity-phase-exponential-growth" title="Link to this heading">#</a></h3>
<p>When <span class="math notranslate nohighlight">\(N \ll K\)</span>, the term <span class="math notranslate nohighlight">\(\bigl(1 - N/K\bigr)\approx 1\)</span>, and:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}  \approx  r(T,M)\,N.
\]</div>
<p>Hence, the <strong>early</strong> dynamics are effectively <strong>exponential</strong>, with the rate</p>
<div class="math notranslate nohighlight">
\[
\mu(T,M)  \approx  r(T,M) 
 \approx 
r_0\,M^{-1/4}\,\exp\!\Bigl(-\tfrac{E}{k_B\,T}\Bigr).
\]</div>
</section>
<section id="late-highdensity-phase-saturation-at-k">
<h3>3.2 Late (High‐Density) Phase: Saturation at <span class="math notranslate nohighlight">\(K\)</span><a class="headerlink" href="#late-highdensity-phase-saturation-at-k" title="Link to this heading">#</a></h3>
<p>As <span class="math notranslate nohighlight">\(N\)</span> approaches <span class="math notranslate nohighlight">\(K\)</span>, resource depletion or waste accumulation slows growth:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}  \to  0
\quad\text{as}\quad
N  \to  K.
\]</div>
<p>Eventually, <span class="math notranslate nohighlight">\(N(t)\)</span> saturates at <span class="math notranslate nohighlight">\(K\)</span>. No matter how <strong>fast</strong> the intrinsic rate <span class="math notranslate nohighlight">\(r(T,M)\)</span> is (e.g., at high temperature for small cells), the population <strong>cannot</strong> surpass the environment’s <strong>carrying capacity</strong>.</p>
</section>
<section id="possible-temperature-size-dependence-of-k">
<h3>3.3 Possible Temperature/Size Dependence of <span class="math notranslate nohighlight">\(K\)</span><a class="headerlink" href="#possible-temperature-size-dependence-of-k" title="Link to this heading">#</a></h3>
<p>In many real ecological or bioprocess contexts, the <strong>carrying capacity</strong> <span class="math notranslate nohighlight">\(K\)</span> might also <strong>depend on temperature</strong> (through effects on nutrient availability, oxygen solubility, etc.) or on <strong>cell size</strong> (larger cells might require more resources, reducing the total cell number sustainable). A simple extension might be:</p>
<div class="math notranslate nohighlight">
\[
K(T,M)
 = 
K_0\,f(T)\,g(M),
\]</div>
<p>where <span class="math notranslate nohighlight">\(f(T)\)</span> and <span class="math notranslate nohighlight">\(g(M)\)</span> are functions that capture how resource utilization or environment capacity scales with temperature and cell size. Then,</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}
 = 
r(T,M)\,N
 \Bigl(1 - \frac{N}{K(T,M)}\Bigr).
\]</div>
<p>Although <strong>analytical</strong> solutions become more complicated, the qualitative behavior remains <strong>logistic</strong>—exponential growth at low <span class="math notranslate nohighlight">\(N\)</span>, saturating at an upper limit.</p>
</section>
</section>
<hr class="docutils" />
<section id="illustrative-equations">
<h2>4. Illustrative Equations<a class="headerlink" href="#illustrative-equations" title="Link to this heading">#</a></h2>
<p>Putting these ideas into a <strong>single</strong> differential equation:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt} =
\bigl[\underbrace{r_0\,M^{-1/4}\,\exp\!\bigl(-E/(k_B\,T)\bigr)}_{\text{Temperature &amp; size dependent rate}}\bigr] 
N(t) 
\Bigl(1 - \frac{N(t)}{K}\Bigr).
\]</div>
<ul class="simple">
<li><p>When <span class="math notranslate nohighlight">\(N(t)\ll K\)</span>, <span class="math notranslate nohighlight">\(dN/dt \approx r_0\,M^{-1/4}\,\exp[-E/(k_B\,T)]  N\)</span>.</p></li>
<li><p>As <span class="math notranslate nohighlight">\(N(t)\to K\)</span>, <span class="math notranslate nohighlight">\(dN/dt\to 0\)</span>.</p></li>
</ul>
<p>Or the more general form if <span class="math notranslate nohighlight">\(K\)</span> also depends on <span class="math notranslate nohighlight">\(T\)</span> and <span class="math notranslate nohighlight">\(M\)</span>:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}
 = 
r(T,M)\,N \Bigl(1 - \frac{N}{\,K(T,M)\!}\Bigr).
\]</div>
</section>
<hr class="docutils" />
<section id="biological-interpretation">
<h2>5. Biological Interpretation<a class="headerlink" href="#biological-interpretation" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>Early Growth Phase (Exponential)</strong></p>
<ul class="simple">
<li><p>Temperature and cell‐size effects dominate, so a <strong>small</strong> microbe at an <strong>optimal</strong> temperature quickly ramps up in population.</p></li>
</ul>
</li>
<li><p><strong>Resource Limitation (Logistic)</strong></p>
<ul class="simple">
<li><p>Eventually, the environment or culture conditions impose a <strong>limit</strong> <span class="math notranslate nohighlight">\(K\)</span>.</p></li>
<li><p>Growth slows down, and the population saturates (e.g., nutrient depletion in a batch culture).</p></li>
</ul>
</li>
<li><p><strong>Role of Cell Size</strong></p>
<ul class="simple">
<li><p><strong>Smaller</strong> cells often have <strong>higher</strong> mass‐specific metabolic rates, thus a <strong>steeper</strong> initial exponential growth. However, if <span class="math notranslate nohighlight">\(K\)</span> is in terms of total <strong>number</strong> of cells, smaller cells might sustain a higher <span class="math notranslate nohighlight">\(K\)</span> (because each cell needs fewer resources). Conversely, if <span class="math notranslate nohighlight">\(K\)</span> is in terms of <strong>total biomass</strong>, the absolute number of larger cells may be lower at carrying capacity.</p></li>
</ul>
</li>
<li><p><strong>Temperature Optima</strong></p>
<ul class="simple">
<li><p>Real systems show an <strong>optimal temperature</strong> for growth. Extremely high <span class="math notranslate nohighlight">\(T\)</span> leads to enzyme denaturation; extremely low <span class="math notranslate nohighlight">\(T\)</span> slows reaction rates.</p></li>
<li><p>Thus, the functional form <span class="math notranslate nohighlight">\(r(T,M)\)</span> can be more complex than a simple Arrhenius. Empirical “Cardinal Temperature Models,” “Ratkowsky Model,” etc., are often used to capture the rising and falling behavior around an optimum.</p></li>
</ul>
</li>
</ol>
</section>
<hr class="docutils" />
<section id="final-summary">
<h2>6. Final Summary<a class="headerlink" href="#final-summary" title="Link to this heading">#</a></h2>
<ul>
<li><p><strong>From Molecular to Population</strong></p>
<ul class="simple">
<li><p>Enzyme kinetics and metabolic rates increase with <strong>temperature</strong> (Arrhenius/Eyring).</p></li>
<li><p><strong>Cell‐level</strong> growth and division rates inherit this temperature dependence and scale with <strong>cell mass</strong> (<span class="math notranslate nohighlight">\(M^{-1/4}\)</span>) per MTE.</p></li>
</ul>
</li>
<li><p><strong>Exponential to Logistic</strong></p>
<ul class="simple">
<li><p>At <strong>low population densities</strong>, growth is effectively <strong>exponential</strong>, with <span class="math notranslate nohighlight">\(\mu(T,M)\)</span>.</p></li>
<li><p>At <strong>high densities</strong>, resources become limiting, and <strong>logistic</strong> growth emerges with a carrying capacity <span class="math notranslate nohighlight">\(K\)</span>.</p></li>
</ul>
</li>
<li><h1 class="rubric" id="integrated-model-frac-dn-dt"><strong>Integrated Model</strong><br />
$$
\frac{dN}{dt}</h1>
<p>\mu(T,M),N
\Bigl(1 - \frac{N}{K}\Bigr)
$<span class="math notranslate nohighlight">\(
(optionally letting \)</span>K = K(T,M)$) succinctly unifies <strong>temperature dependence</strong>, <strong>cell size scaling</strong>, and <strong>density limitations</strong>.</p>
</li>
</ul>
<p>Hence, you get a <strong>temperature‐ and size‐dependent</strong> logistic equation that captures both the <strong>initial exponential ramp</strong> (driven by temperature‐enhanced metabolism in smaller or larger cells) and the <strong>ultimate saturation</strong> governed by the ecological/physical constraints of the environment.</p>
<section id="references">
<h3>References<a class="headerlink" href="#references" title="Link to this heading">#</a></h3>
<ol class="arabic simple">
<li><p>Arrhenius, S. (1889). <em>Über die Reaktionsgeschwindigkeit bei Inversion von Rohrzucker durch Säuren</em>. <strong>Z. Phys. Chem.</strong> 4, 226–248.</p></li>
<li><p>van ’t Hoff, J. H. (1884). <em>Études de dynamique chimique</em>. Frederik Muller &amp; Cie.</p></li>
<li><p>Eyring, H. (1935). The Activated Complex in Chemical Reactions. <strong>J. Chem. Phys.</strong> 3(2), 107–115.</p></li>
<li><p>Polanyi, M. (1935). <strong>Z. Phys. Chem. B</strong> 28, 309–318.</p></li>
<li><p>Johnson, K. A., &amp; Goody, R. S. (2011). The Original Michaelis Constant: Translation of the 1913 Michaelis–Menten Paper. <strong>Biochemistry</strong>, 50(39), 8264–8269.</p></li>
</ol>
<p>Below is a <strong>non‐exhaustive list</strong> of key references and seminal works that underlie the <strong>theoretical</strong> and <strong>empirical</strong> arguments in the derivations above—covering Arrhenius and transition state theory, microbial growth models, and the Metabolic Theory of Ecology.</p>
</section>
</section>
<hr class="docutils" />
<section id="temperature-dependence-of-reaction-rates">
<h2>1. Temperature Dependence of Reaction Rates<a class="headerlink" href="#temperature-dependence-of-reaction-rates" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>Svante Arrhenius</strong> (1889)</p>
<ul class="simple">
<li><p><em>Über die Reaktionsgeschwindigkeit bei Inversion von Rohrzucker durch Säuren.</em><br />
<strong>Zeitschrift für Physikalische Chemie</strong>, <strong>4</strong>, 226–248.</p>
<ul>
<li><p>Seminal paper introducing the <strong>Arrhenius equation</strong> for reaction rates.</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>Henry Eyring</strong> (1935)</p>
<ul class="simple">
<li><p><em>The Activated Complex in Chemical Reactions.</em><br />
<strong>The Journal of Chemical Physics</strong>, <strong>3</strong>, 107–115.</p>
<ul>
<li><p>Foundational work on <strong>Transition State Theory (TST)</strong>, deriving the <strong>Eyring equation</strong>.</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>Keith J. Laidler</strong> (1972)</p>
<ul class="simple">
<li><p><em>The development of the Arrhenius equation.</em><br />
<strong>Journal of Chemical Education</strong>, <strong>39</strong>, 333–336.</p>
<ul>
<li><p>Historical perspective on how the Arrhenius equation came to be.</p></li>
</ul>
</li>
</ul>
</li>
</ol>
</section>
<hr class="docutils" />
<section id="classical-microbial-growth-kinetics">
<h2>2. Classical Microbial Growth Kinetics<a class="headerlink" href="#classical-microbial-growth-kinetics" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>F. Jacob &amp; J. Monod</strong> (1961)</p>
<ul class="simple">
<li><p><em>Genetic regulatory mechanisms in the synthesis of proteins.</em><br />
<strong>Journal of Molecular Biology</strong>, <strong>3</strong>, 318–356.</p>
<ul>
<li><p>While this paper is famously about gene regulation, it lays groundwork for how enzyme kinetics tie into overall cellular processes.</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>E. O. Powell</strong> (1956)</p>
<ul class="simple">
<li><p><em>Growth rate and generation time of bacteria, with special reference to continuous culture.</em><br />
<strong>Journal of General Microbiology</strong>, <strong>15</strong>, 492–511.</p>
<ul>
<li><p>Early treatments of <strong>bacterial division rates</strong> and how they depend on environmental conditions.</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>M. Schaechter, O. Maaløe, &amp; N. O. Kjeldgaard</strong> (1958 &amp; 1962)</p>
<ul class="simple">
<li><p><em>Dependency on medium and temperature of cell size and chemical composition during balanced growth of Salmonella typhimurium.</em><br />
<strong>Journal of General Microbiology</strong>, <strong>19</strong>, 592–606 (1962).</p>
<ul>
<li><p>Classic experimental work linking <strong>temperature</strong> to <strong>cellular physiology</strong> (cell size, composition, growth).</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>J. D. Murray</strong></p>
<ul class="simple">
<li><p><em>Mathematical Biology</em> (various editions).</p>
<ul>
<li><p>Standard reference for <strong>logistic growth</strong>, <strong>exponential growth</strong>, and <strong>population dynamics</strong> modeling.</p></li>
</ul>
</li>
</ul>
</li>
</ol>
</section>
<hr class="docutils" />
<section id="logistic-growth-population-ecology">
<h2>3. Logistic Growth &amp; Population Ecology<a class="headerlink" href="#logistic-growth-population-ecology" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>Pierre F. Verhulst</strong> (1838)</p>
<ul class="simple">
<li><p><em>Notice sur la loi que la population poursuit dans son accroissement.</em><br />
<strong>Correspondance Mathématique et Physique</strong>, <strong>10</strong>, 113–121.</p>
<ul>
<li><p>The original paper introducing the <strong>logistic equation</strong>.</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>R. Pearl &amp; L. J. Reed</strong> (1920)</p>
<ul class="simple">
<li><p><em>On the rate of growth of the population of the United States since 1790 and its mathematical representation.</em><br />
<strong>Proceedings of the National Academy of Sciences (USA)</strong>, <strong>6</strong>, 275–288.</p>
<ul>
<li><p>Classic application of logistic growth to real population data.</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>D. A. Ratkowsky, J. Olley, T. A. McMeekin, &amp; A. Ball</strong> (1982)</p>
<ul class="simple">
<li><p><em>Relationship between temperature and growth rate of bacterial cultures.</em><br />
<strong>Journal of Bacteriology</strong>, <strong>149</strong>, 1–5.</p>
<ul>
<li><p>Introduced what became the <strong>Ratkowsky model</strong>, an empirical expression for microbial growth rates vs. temperature, often used in food microbiology.</p></li>
</ul>
</li>
</ul>
</li>
</ol>
</section>
<hr class="docutils" />
<section id="metabolic-theory-of-ecology-mte">
<h2>4. Metabolic Theory of Ecology (MTE)<a class="headerlink" href="#metabolic-theory-of-ecology-mte" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>G. B. West, J. H. Brown, &amp; B. J. Enquist</strong> (1997)</p>
<ul class="simple">
<li><p><em>A general model for the origin of allometric scaling laws in biology.</em><br />
<strong>Science</strong>, <strong>276</strong>, 122–126.</p>
<ul>
<li><p>Landmark paper deriving the <strong>quarter‐power</strong> scaling (e.g., (M^{3/4}), (M^{-1/4})) based on fractal‐like transport networks.</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>G. B. West, J. H. Brown, &amp; B. J. Enquist</strong> (1999)</p>
<ul class="simple">
<li><p><em>The Fourth Dimension of Life: Fractal Geometry and Allometric Scaling of Organisms.</em><br />
<strong>Science</strong>, <strong>284</strong>, 1677–1679.</p>
<ul>
<li><p>Follow‐up discussion on how scaling laws affect metabolism and other life processes.</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>J. H. Brown, J. F. Gillooly, A. P. Allen, V. M. Savage, &amp; G. B. West</strong> (2004)</p>
<ul class="simple">
<li><p><em>Toward a Metabolic Theory of Ecology.</em><br />
<strong>Ecology</strong>, <strong>85</strong>, 1771–1789.</p>
<ul>
<li><p>Synthesis article outlining how <strong>metabolic rates</strong> scale with <strong>body mass</strong> and <strong>temperature</strong>, forming the basis of <strong>MTE</strong>.</p></li>
</ul>
</li>
</ul>
</li>
</ol>
</section>
<hr class="docutils" />
<section id="integrative-synthetic-works">
<h2>5. Integrative / Synthetic Works<a class="headerlink" href="#integrative-synthetic-works" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>S. A. Levin</strong> (Ed.)** (2009)</p>
<ul class="simple">
<li><p><em>The Princeton Guide to Ecology.</em><br />
<strong>Princeton University Press.</strong></p>
<ul>
<li><p>Contains chapters synthesizing <strong>metabolic theory</strong>, <strong>population dynamics</strong>, and <strong>microbial ecology</strong>.</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>B. Kerr, M. A. Riley, M. W. Feldman, &amp; B. J. M. Bohannan</strong> (2002)</p>
<ul class="simple">
<li><p><em>Local dispersal promotes biodiversity in a real‐life game of rock–paper–scissors.</em><br />
<strong>Nature</strong>, <strong>418</strong>, 171–174.</p>
<ul>
<li><p>While not specifically about temperature, it’s an example of coupling <strong>population dynamics</strong> with <strong>microbial interactions</strong>, illustrating the complexity of real ecological systems.</p></li>
</ul>
</li>
</ul>
</li>
<li><p><strong>Metzler, D. E.</strong> (2001)</p>
<ul class="simple">
<li><p><em>Biochemistry: The Chemical Reactions of Living Cells.</em> 2nd ed., <strong>Academic Press</strong>.</p>
<ul>
<li><p>Comprehensive reference on <strong>enzyme kinetics</strong> in the broader physiological context.</p></li>
</ul>
</li>
</ul>
</li>
</ol>
<hr class="docutils" />
<section id="additional-notes">
<h3>Additional Notes:<a class="headerlink" href="#additional-notes" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>Empirical vs. Theoretical</strong>: Many of the earliest derivations of <strong>temperature‐dependent growth</strong> arose from <strong>empirical fits</strong> (e.g., Arrhenius, Ratkowsky) to experimental microbial data. <strong>Transition State Theory</strong> was largely developed in <strong>physical chemistry</strong> for chemical reactions but has been adapted for <strong>biochemical</strong> and <strong>enzymatic</strong> rate processes.</p></li>
<li><p><strong>MTE</strong>: Extends these ideas to <strong>whole‐organism</strong> (or <strong>whole‐cell</strong>) metabolism, adding the <strong>allometric</strong> component ((M^{3/4}) scaling for metabolic rate).</p></li>
<li><p><strong>Logistic Growth</strong>: Originates in <strong>population ecology</strong> (Verhulst, Pearl &amp; Reed), but it’s widely applicable to <strong>microbial cultures</strong> in a <strong>batch</strong> environment with finite resources.</p></li>
</ul>
<p>Together, these references provide the <strong>theoretical foundation</strong> and <strong>empirical evidence</strong> for deriving and applying <strong>temperature‐ and size‐dependent logistic (and exponential) growth</strong> models in microbial systems.</p>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#introduction-and-historical-context">1. Introduction and Historical Context</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#arrhenius-equation">1.1. Arrhenius Equation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#eyring-equation-transition-state-theory">1.2. Eyring Equation (Transition State Theory)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#application-to-enzyme-kinetics">1.3. Application to Enzyme Kinetics</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#constants-and-helper-functions">2. Constants and Helper Functions</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#interactive-plots">3. Interactive Plots</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#discussion">4. Discussion</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#conclusion">5. Conclusion</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#enzymes-and-the-temperature-dependence-of-reaction-rates">1. Enzymes and the Temperature Dependence of Reaction Rates</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#arrhenius-or-eyringstyle-dependence">1.1 Arrhenius or Eyring‐Style Dependence</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#summation-of-intracellular-reaction-rates">1.2 Summation of Intracellular Reaction Rates</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#from-singlecell-growth-to-division">2. From Single‐Cell Growth to Division</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-division-as-a-stochastic-process">2.1 Cell Division as a (Stochastic) Process</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#average-doubling-time">2.2 Average Doubling Time</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#exponential-population-growth">3. Exponential Population Growth</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#linking-cell-division-rate-to-population-growth">3.1 Linking Cell Division Rate to Population Growth</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#growth-rate-vs-doubling-time">3.2 Growth Rate vs Doubling Time</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#putting-it-all-together">4. Putting It All Together</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#caveats-and-extensions">5. Caveats and Extensions</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#final-schematic-summary">6. Final Schematic Summary</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#temperaturedependent-enzyme-kinetics">1. Temperature‐Dependent Enzyme Kinetics</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">1.1 Arrhenius or Eyring‐Style Dependence</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-division-and-doubling-rate">2. Cell Division and Doubling Rate</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#division-as-a-stochastic-process">2.1 Division as a (Stochastic) Process</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">3. Exponential Population Growth</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#linking-division-rate-to-population-size">3.1 Linking Division Rate to Population Size</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#incorporating-cell-size-scaling-from-metabolic-theory-of-ecology-mte">4. Incorporating Cell Size Scaling from Metabolic Theory of Ecology (MTE)</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#core-premise-of-mte">4.1 Core Premise of MTE</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#growth-rate-from-metabolic-rate">4.2 Growth Rate from Metabolic Rate</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-size-scaling-factor">4.3 Cell Size Scaling Factor</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#exponential-growth-with-mte-scaling">4.4 Exponential Growth with MTE Scaling</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#id3">5. Putting It All Together</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#key-takeaways">6. Key Takeaways</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#recap-temperature-and-cell-size-dependence-in-exponential-growth">1. Recap: Temperature and Cell Size Dependence in Exponential Growth</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#logistic-growth-the-basic-model">2. Logistic Growth: The Basic Model</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#classic-logistic-equation">2.1 Classic Logistic Equation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#temperature-size-dependence-in-logistic-growth">2.2 Temperature &amp; Size Dependence in Logistic Growth</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#how-it-all-ties-together">3. How It All Ties Together</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#initial-lowdensity-phase-exponential-growth">3.1 Initial (Low‐Density) Phase: Exponential Growth</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#late-highdensity-phase-saturation-at-k">3.2 Late (High‐Density) Phase: Saturation at <span class="math notranslate nohighlight">\(K\)</span></a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#possible-temperature-size-dependence-of-k">3.3 Possible Temperature/Size Dependence of <span class="math notranslate nohighlight">\(K\)</span></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#illustrative-equations">4. Illustrative Equations</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#biological-interpretation">5. Biological Interpretation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#final-summary">6. Final Summary</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#references">References</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#temperature-dependence-of-reaction-rates">1. Temperature Dependence of Reaction Rates</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#classical-microbial-growth-kinetics">2. Classical Microbial Growth Kinetics</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#logistic-growth-population-ecology">3. Logistic Growth &amp; Population Ecology</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#metabolic-theory-of-ecology-mte">4. Metabolic Theory of Ecology (MTE)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#integrative-synthetic-works">5. Integrative / Synthetic Works</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#additional-notes">Additional Notes:</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By MulQuaBio collective!
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>