
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>The Metabolic Basis of Ecology and Evolutionary dynamics &#8212; MulQuaBio</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css?v=01a0bdc2" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notebooks/MetabolicBasis';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Single Populations" href="Populations.html" />
    <link rel="prev" title="Biological Computing in R" href="R.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../_static/logo.png" class="logo__image only-light" alt="MulQuaBio - Home"/>
    <script>document.write(`<img src="../_static/logo.png" class="logo__image only-dark" alt="MulQuaBio - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Welcome to The Multilingual Quantitative Biologist!
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Computing</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Unix.html">UNIX and Linux</a></li>
<li class="toctree-l1"><a class="reference internal" href="ShellScripting.html">Shell scripting</a></li>
<li class="toctree-l1"><a class="reference internal" href="Git.html">Version control with Git</a></li>
<li class="toctree-l1"><a class="reference internal" href="LaTeX.html">Scientific documents with <span class="math notranslate nohighlight">\(\LaTeX\)</span></a></li>
<li class="toctree-l1"><a class="reference internal" href="Python.html">Biological Computing in Python</a></li>
<li class="toctree-l1"><a class="reference internal" href="R.html">Biological Computing in R</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Ecological &amp; Evolutionary Modelling</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">The Metabolic Basis of Ecology and Evolutionary dynamics</a></li>
<li class="toctree-l1"><a class="reference internal" href="Populations.html">Single Populations</a></li>
<li class="toctree-l1"><a class="reference internal" href="Interactions.html">Species Interactions</a></li>


<li class="toctree-l1"><a class="reference internal" href="../references.html">References</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Basic Data Analyses and Statistics</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../intro-Stats.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="Data_R.html">Data Management and Visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="ExpDesign.html">Experimental design and Data exploration</a></li>
<li class="toctree-l1"><a class="reference internal" href="t_F_tests.html">Basic hypothesis testing: <span class="math notranslate nohighlight">\(t\)</span> and <span class="math notranslate nohighlight">\(F\)</span> tests</a></li>
<li class="toctree-l1"><a class="reference internal" href="Regress.html">Linear Models: Regression</a></li>
<li class="toctree-l1"><a class="reference internal" href="ANOVA.html">Linear Models: Analysis of variance</a></li>
<li class="toctree-l1"><a class="reference internal" href="MulExpl.html">Linear Models: Multiple explanatory variables</a></li>
<li class="toctree-l1"><a class="reference internal" href="MulExplInter.html">Linear Models: Multiple variables with interactions</a></li>
<li class="toctree-l1"><a class="reference internal" href="ModelSimp.html">Model simplification</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Advanced Data Analyses and Statistics</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="GLMs.html">Generalised Linear Models</a></li>
<li class="toctree-l1"><a class="reference internal" href="NLLS.html">Model Fitting using Non-linear Least-squares</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendices</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Appendix-JupyIntro.html">Introduction to Jupyter</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-Data-Python.html">Data analyses  with Python &amp; Jupyter</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-Maths.html">Mathematical models in Jupyter</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-MathsForBiologists/MathsForBiologists.html">Maths for Biologists</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-Databases.html">Databases <span class="tocSkip"></span></a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-NLLS-Python.html">Model fitting in Python</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-MiniProj.html">The Computing Miniproject</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-Assessment.html">TMQB Coursework Assessment</a></li>
<li class="toctree-l1"><a class="reference internal" href="Appendix-HighPerformanceComputing.html">Introduction to High-Performance Computing (HPC)</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/MulQuaBio/MQB" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/MulQuaBio/MQB/issues/new?title=Issue%20on%20page%20%2Fnotebooks/MetabolicBasis.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/notebooks/MetabolicBasis.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>The Metabolic Basis of Ecology and Evolutionary dynamics</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#introduction">Introduction</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#energy-and-metabolic-rate">Energy and Metabolic Rate</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#importance-of-size">Importance of Size</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#geometric-foundations-of-the-metabolic-scaling-law">Geometric foundations of the metabolic scaling law</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#biological-implications-of-the-geometric-scaling-constraint">Biological implications of the geometric scaling constraint</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#going-beyond-geometric-constraints">Going beyond geometric constraints</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#importance-of-temperature-and-thermodynamic-constraints">Importance of Temperature (and Thermodynamic constraints)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#classical-transition-state-theory-tst">Classical Transition State Theory (TST)</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overall-temperature-dependence-of-the-rate">Overall Temperature Dependence of the Rate</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#putting-it-together-temperature-size-scaling-and-metabolic-rate">Putting it together: temperature, size scaling, and metabolic rate</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#ecological-implications">Ecological implications</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#from-intracellular-reaction-rates-to-cell-growth">From intracellular reaction rates to cell growth</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#from-singlecell-growth-to-division">From single‐cell growth to division</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-division-is-a-stochastic-process">Cell division is a stochastic process</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#average-doubling-time">(Average) Doubling time</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#from-cell-division-rate-to-population-growth">From cell division rate to population growth</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#growth-rate-vs-doubling-time">Growth Rate vs Doubling Time</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#incorporating-cell-size-scaling">Incorporating Cell Size Scaling</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#a-more-general-relationship">A more general relationship</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#from-exponential-to-logistic-growth">From exponential to logistic growth</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#classic-logistic-equation">Classic Logistic Equation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#temperature-size-dependence-in-logistic-growth">2.2 Temperature &amp; Size Dependence in Logistic Growth</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#temperature-and-size-dependence-of-k">Temperature and Size Dependence of <span class="math notranslate nohighlight">\(K\)</span></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#population-dynamics">Population dynamics</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#biological-interpretations">Biological Interpretations</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#readings">Readings</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="the-metabolic-basis-of-ecology-and-evolutionary-dynamics">
<h1>The Metabolic Basis of Ecology and Evolutionary dynamics<a class="headerlink" href="#the-metabolic-basis-of-ecology-and-evolutionary-dynamics" title="Link to this heading">#</a></h1>
  <!-- * Biochemical kinetics and thermodynamics
  * Scaling laws - space and time
  * Implications for ecological and evolutionary dynamics
  * Examples & Applications   --><blockquote class="epigraph">
<div><p>The struggle for existence of living beings is not for the fundamental constituents of food, but for the possession of the free energy obtained, chiefly by means of the green plant, from the transfer of radiant energy from the hot sun to the cold earth.</p>
<p class="attribution">—Ludwig Boltzmann, 1886, “<em>The Second Law of Thermodynamics</em>”
</p>
</div></blockquote>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Link to this heading">#</a></h2>
<p>Metabolism is fundamental to the understanding of ecological and evolutionary processes.</p>
<p>The laws of thermodynamics are fundamental principles describing how energy and heat behave in physical, chemical, and living systems. As Boltzmann noted <span id="id1">[<a class="reference internal" href="../references.html#id5" title="Ludwig Boltzmann. The second law of thermodynamics. Naturwissenschaftliche Rundschau, 1(1):293–297, 1886. Translated and cited in later works on thermodynamics and biological processes. Original in German.">Boltzmann, 1886</a>]</span>, the <em>Second Law of Thermodynamics</em> is fundamental for understanding how biological cells and systems work as well as their ecological and evolutionary dynamics:</p>
<blockquote>
<div><p>In any <em>spontaneous</em> process, the total entropy of an <em>isolated</em> system can never decrease. Equivalently, it can stay the same (in a reversible process) or increase (in an irreversible process).</p>
</div></blockquote>
<p>The two direct implications of this Law are directly relevant to biological systems:</p>
<ul class="simple">
<li><p>The first implication is the <em>“arrow of time”</em> — natural processes tend to move toward greater Entropy (disorder). What makes life and biological processes unique (in contrast to purely physical or chemical processes) is that in essence, the purpose of living organisms is to <em>counteract</em> this law by producing order (living biomass) from disorder (photons and molecules). At the most fundamental level, this is possible because living organisms are <em>not isolated</em> — thay are open systems that continuously exchange energy and matter with their surroundings.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>While organisms maintain or increase their internal order (i.e., decrease entropy locally) by building and organizing complex molecules, they do so by taking in energy from external sources (e.g., sunlight or food) and releasing waste heat and byproducts back into their environment. This flow of energy and matter ultimately does increase the total entropy of the universe. Thus, life’s local decrease in entropy comes at the expense of a larger increase in entropy elsewhere, in accordance with the second law.</p>
</div>
<ul class="simple">
<li><p>The second implication is that it is impossible to convert heat completely into work without waste (there is no perfect engine) — living organisms like any engine are subject to this limitation, but are typically designed by natural selection to maximise or optimise their energy conversion efficiency. Indeed, Lotka <span id="id2">[<a class="reference internal" href="../references.html#id6" title="Alfred J Lotka. Contribution to the energetics of evolution. Proceedings of the National Academy of Sciences, 8(6):147–151, 1922.">Lotka, 1922</a>]</span> made this observation over a 100 years ago:</p></li>
</ul>
<blockquote>
<div><p><em>It has been pointed out by Boltzmann’ that the fundamental object ofcontention in the life-struggle, in the evolution of the organic world, is available energy. In accord with this observation is the principle that,in the struggle for existence, the advantage must go to those organismswhose energy-capturing devices are most efficient in directing available energy into channels favorable to the preservation of the species.</em></p>
</div></blockquote>
</section>
<section id="energy-and-metabolic-rate">
<h2>Energy and Metabolic Rate<a class="headerlink" href="#energy-and-metabolic-rate" title="Link to this heading">#</a></h2>
<p>Energy is a measurable property that quantified an object’s capacity to perform work (in a physical sense). In living organisms, a key function is to harness and utilize energy to support growth, maintenance, and reproduction. Plants and other autotrophs capture photons through photosynthesis, transforming light energy into chemical energy stored in glucose. Heterotrophs, such as animals, obtain energy by breaking chemical bonds in organic molecules. A crucial molecule in this process is ATP (adenosine triphosphate), which serves as a primary energy carrier within cells and powers numerous biochemical reactions.</p>
<p><em>Metabolism</em>, the set of chemical reactions within an organism, determines <em>metabolic rate</em>, which is the rate of energy use, often measured in J/s, kcal/day, or Watts. Organisms must balance energy consumption and expenditure:</p>
<div class="math notranslate nohighlight">
\[
    \text{Energy Consumed} = \text{Energy Used (Maintenance + Growth + Movement)}.
\]</div>
<p>Metabolic rates influence key life-history traits, including movement rates, development speed, lifespan, and reproductive output <span id="id3">[<a class="reference internal" href="../references.html#id4" title="Anthony I Dell, Samraat Pawar, and Van M Savage. Systematic variation in the temperature dependence of physiological and ecological traits. Proceedings of the National Academy of Sciences, 108(26):10591–10596, 2011.">Dell <em>et al.</em>, 2011</a>]</span>.</p>
<p>The metabolic theory of ecology (MTE) <span id="id4">[<a class="reference internal" href="../references.html#id3" title="J H Brown, J F Gillooly, A P Allen, V M Savage, and G B West. Toward a metabolic theory of ecology. Ecology, 85(7):1771–1789, 2004. URL: isi:000223113500001, doi:10.1890/03-9000.">Brown <em>et al.</em>, 2004</a>]</span> provides a framework for understanding how metabolic processes scale across levels of biological organization and influence ecological dynamics. It combines the <em>biomechanical</em> constraints of cell or body size of organisms and the <em>thermodynamic</em> effects of temperature on biological reaction kinetics into one equation for organism-level (that of single, integral organisms, either unicells or multi-cells) metabolic rate.</p>
<p>We will now seperately consider the biophysical effects of size and temperature and then combine them into a unified understanding (and equations).</p>
</section>
<section id="importance-of-size">
<h2>Importance of Size<a class="headerlink" href="#importance-of-size" title="Link to this heading">#</a></h2>
<p>Metabolic rate (<span class="math notranslate nohighlight">\(B\)</span>) increases with body size (<span class="math notranslate nohighlight">\(M\)</span>) according to the allometric scaling law <span id="id5">[<a class="reference internal" href="../references.html#id2" title="M. Kleiber. Body size and metabolic rate. Physiol. Rev., 27(4):511–541, 1947. URL: http://www.planta.cn/forum/files{\_}planta/511{\_}131.pdf{\%}5Cnpapers2://publication/uuid/008695C4-AE04-48B5-A89C-813ABFD00CB0, doi:10.1152/physrev.1947.27.4.511.">Kleiber, 1947</a>]</span>:</p>
<div class="math notranslate nohighlight">
\[
    B = B_0 M^b,
\]</div>
<p>or equivalently:</p>
<div class="math notranslate nohighlight">
\[
    \log_{10}(B) = \log_{10}(B_0) + b \log_{10}(M),
\]</div>
<p>where <span class="math notranslate nohighlight">\(b\)</span> is typically close to 3/4 for multicellular euykaryotic organisms. This implies that larger organisms have higher absolute metabolic rates but lower mass-specific rates (<span class="math notranslate nohighlight">\(B/M\)</span>):</p>
<div class="math notranslate nohighlight">
\[
    \frac{B}{M} = B_0 M^{b-1}.
\]</div>
<p>For example, a mouse (0.1 kg) needs approximately 12.3 kcal/day, while its mass-specific rate is 123 kcal/(kg<span class="math notranslate nohighlight">\(\cdot\)</span>day).</p>
<p>This reduction in per-mass energy use with size arises from a combination of constraints - geometric, overheating risk (Refs), and within-organism resource distribution (West et al., 2005).</p>
<section id="geometric-foundations-of-the-metabolic-scaling-law">
<h3>Geometric foundations of the metabolic scaling law<a class="headerlink" href="#geometric-foundations-of-the-metabolic-scaling-law" title="Link to this heading">#</a></h3>
<p>Historically, a geometric argument was used to suggest that metabolic rate scales like the surface area, which in turn scales as the <span class="math notranslate nohighlight">\(\tfrac{2}{3}\)</span> power of mass.</p>
<p>Let us assume an organism (or cell) is <em>approximately spherical</em> of radius <span class="math notranslate nohighlight">\(l\)</span>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>We will use <span class="math notranslate nohighlight">\(l\)</span> instead of the traditional <span class="math notranslate nohighlight">\(l\)</span> to denote the radius of a sphere to avoid confusion, because the latter is a comonlyu used symbol for population growth rate (which we will encounter in the next chapter).</p>
</div>
<p>Let’s start with the basic equations:</p>
<p><em>Surface area of sphere</em> <span class="math notranslate nohighlight">\(A = 4\pi l^2\)</span></p>
<p><em>Volumev of a sphere</em> <span class="math notranslate nohighlight">\(V = \tfrac{4}{3}\pi l^3\)</span></p>
<p>If the organism/cell has a <em>constant density</em> <span class="math notranslate nohighlight">\(\rho\)</span>, then its mass <span class="math notranslate nohighlight">\(m\)</span> is proportional to its volume:</p>
<div class="math notranslate nohighlight">
\[
    m \propto V \propto l^3 \quad \Longrightarrow \quad l \propto m^{1/3}.
\]</div>
<p>Then, the <em>surface-to-volume ratio</em> is</p>
<div class="math notranslate nohighlight">
\[
    \frac{A}{V} 
    = \frac{4\pi l^2}{\tfrac{4}{3}\pi l^3} 
    = \frac{3}{l}.
\]</div>
<p><em>That is, as <span class="math notranslate nohighlight">\(l\)</span> increases, <span class="math notranslate nohighlight">\(\tfrac{A}{V}\)</span> decreases (<span class="math notranslate nohighlight">\(\propto 1/l\)</span>).</em></p>
<p>Now, in  biological cells, the <em>surface area</em> is where exchange of nutrients, oxygen, and waste takes place, while the <em>volume</em> represents total metabolic demand (more “living material” inside).</p>
<p>So, <em>larger cell radius</em> <span class="math notranslate nohighlight">\(\Rightarrow\)</span> <em>lower surface-to-volume ratio</em> <span class="math notranslate nohighlight">\(\Rightarrow\)</span> <em>harder to meet the metabolic needs</em> of the cell or organism.</p>
<p>This leads to the classic geometric argument that metabolic rate <span class="math notranslate nohighlight">\(B\)</span> (e.g., oxygen consumption per unit time) is limited by the total surface available for exchange:</p>
<div class="math notranslate nohighlight">
\[
    B \propto A.
\]</div>
<p>Using <span class="math notranslate nohighlight">\(A = 4\pi l^2\)</span>, we get</p>
<div class="math notranslate nohighlight">
\[
    B \propto l^2.
\]</div>
<p>Since <span class="math notranslate nohighlight">\(m \propto l^3\)</span>, we have <span class="math notranslate nohighlight">\(l \propto m^{1/3}\)</span>. Therefore,</p>
<div class="math notranslate nohighlight">
\[
A \propto l^2 
\propto (m^{1/3})^2 
\propto m^{2/3}.
\]</div>
<p>Hence,</p>
<div class="math notranslate nohighlight">
\[
    B \propto m^{2/3}.
\]</div>
<p>Including a scaling constant upo front,</p>
<div class="math notranslate nohighlight">
\[
    B = B_0  m^{\frac{2}{3}},
\]</div>
<p>which is the same as the first scaling equation above with <span class="math notranslate nohighlight">\(b = \frac{2}{3}\)</span>.</p>
<p>That is, this scaling arises from the fgact that an organism’s metabolic rate scales with its ability to exchange materials through a surface (which goes as <em>length squared</em>), while its mass (and thus total metabolic demand) goes as <em>length cubed</em>.</p>
</section>
<section id="biological-implications-of-the-geometric-scaling-constraint">
<h3>Biological implications of the geometric scaling constraint<a class="headerlink" href="#biological-implications-of-the-geometric-scaling-constraint" title="Link to this heading">#</a></h3>
<p>Thus, small cells have a <em>relatively large surface area</em> for their volume. They can exchange nutrients and wastes quickly, supporting higher metabolic rates <em>per unit volume</em>. In contrast, larger cells (or organisms) have a <em>relatively smaller surface area</em> per unit volume. Consequently, they may be limited in how fast they can exchange vital substances, placing a cap on metabolic rate <em>per unit mass</em> or <em>per unit volume</em>. This is why cells are relatively small: They need <em>large surface area relative to volume</em> to ensure adequate exchange of nutrients/oxygen and removal of wastes for the metabolic demands of their volume.</p>
</section>
</section>
<section id="going-beyond-geometric-constraints">
<h2>Going beyond geometric constraints<a class="headerlink" href="#going-beyond-geometric-constraints" title="Link to this heading">#</a></h2>
<p>Th above surface-to-volume geo biophysical constraint indeed holds at the level of cells, but is not the full story.</p>
<p>Empirically, the scaling most multicellular organisms in fact follows a somewhat different exponent (<span class="math notranslate nohighlight">\(\approx \frac{3}{4}\)</span>, known as <em>Kleiber’s law</em>. This comes from the two additonal constraints: the requirement of heat dissipation and energy distribution (circulation) thorughout the body. We will not cover these here, but if you are intersted, please have a look at the readings at the end of this chapter.</p>
</section>
<section id="importance-of-temperature-and-thermodynamic-constraints">
<h2>Importance of Temperature (and Thermodynamic constraints)<a class="headerlink" href="#importance-of-temperature-and-thermodynamic-constraints" title="Link to this heading">#</a></h2>
<p>Temperature profoundly impacts metabolism, as biochemical reaction rates in cells increase <em>inexorably</em> with temperature following the Boltzmann-Arrhenius equation:</p>
<div class="math notranslate nohighlight">
\[
    k = A e^{-\frac{E_a}{k_B T}},
\]</div>
<p>where <span class="math notranslate nohighlight">\(k\)</span> is the reaction rate, <span class="math notranslate nohighlight">\(k_0\)</span> is a constant, <span class="math notranslate nohighlight">\(E_a\)</span> is the activation energy, <span class="math notranslate nohighlight">\(k_B\)</span> is the Boltzmann constant, and <span class="math notranslate nohighlight">\(T\)</span> is temperature in Kelvin. This relationship explains the characteristic thermal performance curve observed in biological processes, where rates increase with temperature to an optimum before declining due to enzyme denaturation or other stress factors (Johnson et al., 1974) (which we will consider later below).</p>
<p>The <em>Arrhenius equation</em> and its variants (often collectively referred to as the “Boltzmann-Arrhenius” equation in certain contexts) are cornerstones of chemical kinetics. These formulations describe how reaction rates vary with temperature. For <em>enzyme kinetics</em>, the temperature dependence can also be described via modified forms of the Arrhenius equation, sometimes combined with effects such as protein denaturation at higher temperatures.</p>
<p>Historically, this work can be traced back to <em>Jacobus Henricus van ‘t Hoff</em> (1852–1911), whose studies first quantified the temperature dependence of reaction rates. Building on van ‘t Hoff’s work, <em>Svante Arrhenius</em> (1859–1927) proposed a more explicit relationship that connected the rate constant <span class="math notranslate nohighlight">\(k\)</span> of a chemical reaction to an exponential function of temperature:</p>
<div class="math notranslate nohighlight">
\[
    k = A \exp\left( -\frac{E_a}{RT} \right),
\]</div>
<p>where:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\( A \)</span> is the pre-exponential factor (or frequency factor),</p></li>
<li><p><span class="math notranslate nohighlight">\( E_a \)</span> is the activation energy,</p></li>
<li><p><span class="math notranslate nohighlight">\( R \)</span> is the universal gas constant, and</p></li>
<li><p><span class="math notranslate nohighlight">\( T \)</span> is the absolute temperature in Kelvin.</p></li>
</ul>
<p>For <em>enzyme-catalyzed reactions</em>, temperature significantly affects the reaction velocity <span class="math notranslate nohighlight">\(v\)</span> or the turnover number (<em>catalytic rate constant</em> <span class="math notranslate nohighlight">\(k_{\mathrm{cat}}\)</span>). At “moderate” temperatures, higher temperature generally increases reaction rates, consistent with Arrhenius behavior. However, at higher temperatures, <em>thermal denaturation</em> of the enzyme reduces or abolishes its catalytic activity. Hence, more realistic models for enzyme kinetics must account for both the <em>Arrhenius increase in rate</em> and the <em>loss of enzyme activity</em> due to denaturation, which we will consider later below.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All organisms on earth are dependent on temperature directly or indirectly: <em>Poikilotherms</em> such as unicells and tiny insects cannot thermoregulate much; <em>Ectotherms</em> such as plants, insects and reptiles, rely on external temperature to regulate their metabolic processes (thermoregulation); <em>Endotherms</em>, such as mammals and birds, generate internal heat to maintain a stable body temperature, but this comes at a significant energetic cost, and ultimately are reliant on the energy harnessed by poikilotherms and ectotherms (starting with green plants).</p>
</div>
<p>The rate of an <em>enzyme-catalyzed reaction</em> can be modelled as:</p>
<div class="math notranslate nohighlight">
\[\displaystyle \text{Rate} = k_\text{cat}(T) \times [\text{C}] \times f([\text{S}]),\]</div>
<p>where,</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(k_\text{cat}(T)\)</span> is strongly dependent on <span class="math notranslate nohighlight">\(T\)</span>,</p></li>
<li><p><span class="math notranslate nohighlight">\([\text{C}]\)</span> is the enzyme concentration,</p></li>
<li><p><span class="math notranslate nohighlight">\(f([\text{S}])\)</span> is a function of substrate concentration (such as the Michaelis-Menten form <span class="math notranslate nohighlight">\(\tfrac{[\text{S}]}{K_M + [\text{S}] }\)</span>).</p></li>
</ul>
<p>We can model <span class="math notranslate nohighlight">\(k_\text{cat}(T)\)</span> using either the <em>Arrhenius</em> or the <em>Eyring</em> Equations. However, because real enzymes <em>denature</em> at higher temperatures, we typically observe a <em>bell-shaped</em> temperature activity profile.</p>
</section>
<section id="classical-transition-state-theory-tst">
<h2>Classical Transition State Theory (TST)<a class="headerlink" href="#classical-transition-state-theory-tst" title="Link to this heading">#</a></h2>
<p>We will now look at <em>classical transition state theory (TST)</em>, which provides a <em>microscopic rationale</em> for the temperature dependence of reaction rates, explaining both <em>how energy barriers (enthalpy)</em> slow down reactions and how <em>changes in molecular order (entropy)</em> can further modulate the speed at which reactants cross from the well of the reactants to the well of reaction products.</p>
<p>TST was developed in the 1930s by <em>Henry Eyring</em> (1901–1981) and others (in particular Meredith Evans and Michael Polanyi).</p>
<p>The key idea underlying the TST is that a chemical reaction proceeds through a high-energy configuration of the reactants called the <em>transition state</em> (or activated complex). Once formed, the transition state either proceeds forward to products or reverts back to reactants, adhering to two key assumptions:</p>
<ul class="simple">
<li><p>Reagents must first form an <em>activated complex</em>, which is in a “quasi-equilibrium” with the reactants.</p></li>
<li><p>The rate of the reaction then depends on how frequently and efficiently this activated complex “crosses over” from the reactant side of the potential energy surface to the product side.</p></li>
</ul>
<p>Using statistical mechanics arguments, TST yields the famous <em>Eyring (or Eyring–Polanyi) equation</em>, sometimes called the “absolute rate equation:</p>
<div class="math notranslate nohighlight">
\[
    k = \frac{k_B T}{h} \exp\Bigl(\tfrac{\Delta S^\ddagger}{R}\Bigr)\exp\Bigl(-\tfrac{\Delta H^\ddagger}{RT}\Bigr).
\]</div>
<p>where</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(k\)</span> is the rate constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(k_B\)</span> is Boltzmann’s constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(h\)</span> is Planck’s constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(R\)</span> is the gas constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(T\)</span> is the absolute temperature,</p></li>
<li><p><span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> is the <em>activation enthalpy</em>,</p></li>
<li><p><span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> is the <em>activation entropy</em>.</p></li>
</ul>
<p>The key result of TST is the <em>Eyring</em> (or <em>Eyring-Polanyi</em>) Equation, usually written as:</p>
<div class="math notranslate nohighlight">
\[
    \displaystyle k = \kappa \frac{k_B    T}{h} \exp\Bigl( \frac{\Delta S^\ddagger}{R}\Bigr) \exp\Bigl( - \frac{\Delta H^\ddagger}{RT} \Bigr),
\]</div>
<p>where</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(\kappa\)</span> is the transmission coefficient (often assumed to be ~1 for many simple reactions),</p></li>
<li><p><span class="math notranslate nohighlight">\(k_B\)</span> is the Boltzmann constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(h\)</span> is Planck’s constant,</p></li>
<li><p><span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> is the entropy of activation,</p></li>
<li><p><span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> is the enthalpy of activation, and</p></li>
<li><p><span class="math notranslate nohighlight">\(R\)</span> and <span class="math notranslate nohighlight">\(T\)</span> are as before.</p></li>
</ul>
<p>This equation is more often called the “<em>Eyring equation</em>” and offers a more explicitly thermodynamics-based perspective. It reduces to an Arrhenius-like form when you group constants appropriately.</p>
<p>Through this equation, TST provides a <em>molecular-level interpretation</em> for the pre-exponential factor</p>
<p><span class="math notranslate nohighlight">\(\frac{k_B T}{h} \exp \Delta S^\ddagger / R)\)</span>) and the exponential barrier term <span class="math notranslate nohighlight">\(\exp(-\Delta H^\ddagger / (RT))\)</span>) in the rate expression.</p>
<p><span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> is the difference in <em>enthalpy</em> (heat content) between the transition state (activated complex) and the reactants. It represents the <em>energy barrier</em> that must be surmounted for the reaction to occur. A larger <span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> means a higher barrier, which tends to <em>slow down</em> the reaction, especially at lower temperatures.</p>
<p>In the factor <span class="math notranslate nohighlight">\(\exp(-\Delta H^\ddagger / (RT))\)</span>, <span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> appears in the <em>exponential</em> term with a <span class="math notranslate nohighlight">\(1/T\)</span> dependence.Consequently, the <em>higher</em> the activation enthalpy, the <em>more strongly</em> the rate constant depends on temperature (i.e., higher sensitivity in an Arrhenius-type plot).</p>
<p>Next, <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> is the difference in <em>entropy</em> between the transition state and the reactants. It reflects the <em>change in molecular disorder</em> when going from reactants to the activated complex. A <em>positive</em> <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> implies that forming the transition state is relatively more disordered than the reactants, favoring the reaction (larger rate constant). A <em>negative</em> <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> implies a more ordered transition state, which lowers the rate constant by decreasing the pre-exponential factor.</p>
<p>Finally, <span class="math notranslate nohighlight">\(\exp \Delta S^\ddagger / R)\)</span> <em>multiplies</em> the basic frequency factor <span class="math notranslate nohighlight">\(\frac{k_B T}{h}\)</span>. Therefore, even if <span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> is moderate, a large <em>positive</em> <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> can significantly enhance the reaction rate, whereas a negative <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span> can suppress it.</p>
<section id="overall-temperature-dependence-of-the-rate">
<h3>Overall Temperature Dependence of the Rate<a class="headerlink" href="#overall-temperature-dependence-of-the-rate" title="Link to this heading">#</a></h3>
<p>By combining <span class="math notranslate nohighlight">\(\Delta H^\ddagger\)</span> and <span class="math notranslate nohighlight">\(\Delta S^\ddagger\)</span>, TST clarifies <em>how and why</em> reaction rates depend on temperature:</p>
<div class="math notranslate nohighlight">
\[
k(T) = \frac{k_B T}{h} \exp\Bigl(\tfrac{\Delta S^\ddagger}{R}\Bigr) \exp\Bigl(-\tfrac{\Delta H^\ddagger}{RT}\Bigr).
\]</div>
<p>The <em><span class="math notranslate nohighlight">\(\exp(-\Delta H^\ddagger / (RT))\)</span></em> term governs the <em>primary exponential sensitivity</em> of <span class="math notranslate nohighlight">\(k\)</span> to changes in temperature. The <em><span class="math notranslate nohighlight">\(\exp\)</span>\Delta S^\ddagger / R)$</em> term influences the <em>intrinsic magnitude</em> of the rate constant at any given temperature by encoding the “positional” or “configurational” freedom of the transition state.</p>
<p>Experimentalists often plot <span class="math notranslate nohighlight">\(\ln(k/T)\)</span> versus <span class="math notranslate nohighlight">\(1/T\)</span>, known as an <em>Eyring plot</em>:</p>
<div class="math notranslate nohighlight">
\[
    \ln \Bigl(\tfrac{k}{T}\Bigr) = -\frac{\Delta H^\ddagger}{R} \cdot \frac{1}{T} + \frac{\Delta S^\ddagger}{R} + \text{constant}.
\]</div>
<!-- In this plot, The *slope* is $-\Delta H^\ddagger / R$, and the *intercept* is $1/T = 0$ (or the "vertical shift") is $\Delta S^\ddagger / R$.   

Hence, one can directly extract $\Delta H^\ddagger$ and $\Delta S^\ddagger$ from measured rate data at different temperatures to confirm or test validity of the model and the parameter estimates.


## 5. Summary of Key Points

1. *TST Framework*  
   - The reaction proceeds via an activated complex at the top of an energy barrier.  
   - Rate = flux of reactants crossing that barrier.

2. *Activation Enthalpy, $\Delta H^\ddagger$*  
   - Measures the *height* of the energy barrier.  
   - Appears in the $\exp(-\Delta H^\ddagger / (RT))$ factor, dictating strong temperature dependence.

3. *Activation Entropy, $\Delta S^\ddagger$*  
   - Reflects the *ordering/disordering* when forming the transition state.  
   - Enters as $\exp$\Delta S^\ddagger / R)$, setting the magnitude of the frequency factor.

4. *Temperature Dependence*  
   - The Eyring equation, arising from TST, captures both the *enthalpic* (energy) and *entropic* (disorder) contributions to the rate constant's variation with temperature.  

<!-- ## 4. Discussion

1. *Arrhenius vs. Eyring*  
   - The *Arrhenius Equation* is an empirical formula that captures the exponential increase of reaction rate with temperature.  
   - The *Eyring Equation* derives from *Transition State Theory*, relating rate constants to thermodynamic parameters of the transition state ($\Delta H^\ddagger$, $\Delta S^\ddagger$), as depicted in panels (b) and (c) of the figure.

2. *Role of Denaturation*  
   - Both models in this notebook include a *logistic* denaturation factor to reduce active enzyme at high temperatures. Real systems can have more intricate denaturation kinetics, but this simple approach captures the key idea: above a certain threshold, enzyme function declines rapidly.

3. *Bell-Shaped Curve*  
   - Combining an *Arrhenius- or Eyring-like increase* at moderate temperatures with *denaturation* at higher temperatures typically yields the *bell-shaped* temperature-activity curve depicted in many enzyme-kinetics references.  
   - As shown in panel (d), enzymatic rates may also be viewed in the context of substrate concentration and Michaelis–Menten parameters, but the main driver at high temperature is denaturation.

4. *Parameter Interpretation*  
   - *Arrhenius*: $E_a$ (J/mol) is the activation energy barrier.  
   - *Eyring*: $\Delta H^\ddagger$ (J/mol) is the enthalpy of activation; $\Delta S^\ddagger$ (J/(mol·K)) is the entropy of activation. A large positive $\Delta S^\ddagger$ can lead to higher rates, while a large positive $\Delta H^\ddagger$ lowers the rate.

5. *Historical Significance*  
   - *Arrhenius* (1889) built on *van 't Hoff's* work on temperature dependence of equilibrium constants and reaction rates.  
   - *Henry Eyring* (1935) developed Transition State Theory, leading to the *Eyring-Polanyi equation*, providing a more fundamental theoretical underpinning for temperature-dependent rate processes in chemistry and biochemistry. --><!-- The combination of these models, especially with the logistic denaturation factor, nicely illustrates the *bell-shaped* temperature dependence typically observed in real enzymes. Such analyses can be extended to full *Michaelis–Menten* kinetics and to distinguishing *enthalpic* vs. *entropic* components of catalysis. (Knapp & Huang 2022) --></section>
</section>
<section id="putting-it-together-temperature-size-scaling-and-metabolic-rate">
<h2>Putting it together: temperature, size scaling, and metabolic rate<a class="headerlink" href="#putting-it-together-temperature-size-scaling-and-metabolic-rate" title="Link to this heading">#</a></h2>
<p>Thus, metabolism drives growth rates and influences population dynamics through its scaling with size and temperature (Brown et al., 2004). Larger organisms exhibit slower per-mass metabolic rates but greater absolute energy use (Savage et al., 2004). Temperature affects metabolic rates exponentially, shaping life-history strategies and population dynamics (Dell et al., 2011).</p>
<p>Combining these gives MTE’s fundamental equation:</p>
<div class="math notranslate nohighlight">
\[
B = B'_0 m^b e^{-\frac{E_a}{k_BT}}
\]</div>
<p>Where <span class="math notranslate nohighlight">\(B'_0\)</span> is a <em>size-</em> and <em>temperature-independent</em> (normalization) constant.</p>
</section>
<section id="ecological-implications">
<h2>Ecological implications<a class="headerlink" href="#ecological-implications" title="Link to this heading">#</a></h2>
<p>We can now consider how the temperature dependence of <em>enzyme kinetics</em> and basic <em>cellular physiology</em> leads to a <em>temperature‐dependent</em> rate of <em>cell division</em>, and ultimately to <em>exponential population growth</em>. We will use microbes as a model because of the relative simplicity of their growth process, which relies on simple cell division.</p>
<section id="from-intracellular-reaction-rates-to-cell-growth">
<h3>From intracellular reaction rates to cell growth<a class="headerlink" href="#from-intracellular-reaction-rates-to-cell-growth" title="Link to this heading">#</a></h3>
<p>A living cell’s <em>growth and division</em> require a <em>multitude of coupled enzyme reactions</em> (for nutrient uptake, biosynthesis, energy generation, etc.). MTE assumes that <em>one key rate‐limiting step</em> (or a small set of them) dominates the overall pace of cell growth and division. If those rate‐limiting reactions follow Arrhenius‐type kinetics, then the <em>cellular physiological rate</em> (e.g., biomass production rate, cell cycle progression rate) will inherit a similar temperature dependence:</p>
<div class="math notranslate nohighlight">
\[
    \text{Physiological rate}  \propto  k(T)  \approx  A  \exp\Bigl(-\tfrac{E_a}{R  T}\Bigr).
\]</div>
<p>Hence, as <span class="math notranslate nohighlight">\(T\)</span> increases (within a tolerable range), <em>enzyme‐catalyzed processes speed up</em>, and so does the <em>cell’s overall metabolism</em> and ability to replicate.</p>
</section>
<section id="from-singlecell-growth-to-division">
<h3>From single‐cell growth to division<a class="headerlink" href="#from-singlecell-growth-to-division" title="Link to this heading">#</a></h3>
<section id="cell-division-is-a-stochastic-process">
<h4>Cell division is a stochastic process<a class="headerlink" href="#cell-division-is-a-stochastic-process" title="Link to this heading">#</a></h4>
<p>To move from enzyme kinetics to <em>cell‐level</em> dynamics every cell must <em>Replicate all essential components</em> (DNA, proteins, membranes, etc.), Reach a certain “completion threshold” in biomass or chromosome replication, and <em>Divide</em> into two daughter cells.</p>
<p>Because cellular biochemical events occur stochastically (randomly), the <em>exact</em> division time of any one cell can vary. However, in an <em>exponential (log) growth phase</em> with abundant nutrients, the <em>average</em> cell‐division rate <span class="math notranslate nohighlight">\(\lambda\)</span> (divisions per unit time) is fairly constant (Jafarpour et al 2020). This average division rate is ultimately <em>tied</em> to the cumulative rate of intracellular biochemical and biosynthetic reactions, and thus inherits their temperature dependence.</p>
</section>
<section id="average-doubling-time">
<h4>(Average) Doubling time<a class="headerlink" href="#average-doubling-time" title="Link to this heading">#</a></h4>
<p>Let <span class="math notranslate nohighlight">\(\tau_d\)</span> be the <em>average doubling time</em> (the mean time for one cell to become two). Then,</p>
<div class="math notranslate nohighlight">
\[
\lambda  =  \frac{1}{\tau_d} 
\Longleftrightarrow  
\tau_d  =  \frac{1}{\lambda}.
\]</div>
<p>From the <em>Arrhenius perspective</em>, we can write</p>
<div class="math notranslate nohighlight">
\[
     \tau_d(T) \approx \tau_{d,0}  \exp\Bigl(\tfrac{E_a}{R  T}\Bigr),
\]</div>
<p>so that the cell’s doubling time <em>decreases</em> with <em>increasing</em> temperature (again, within the physiological limit). Equivalently, the <em>division rate</em> <span class="math notranslate nohighlight">\(\lambda(T) = 1/\tau_d(T)\)</span> would <em>increase</em> with temperature:</p>
<div class="math notranslate nohighlight">
\[
    \lambda(T)  \approx  \lambda_0   \exp\Bigl(-\tfrac{E_a}{R T}\Bigr).
\]</div>
</section>
</section>
<section id="from-cell-division-rate-to-population-growth">
<h3>From cell division rate to population growth<a class="headerlink" href="#from-cell-division-rate-to-population-growth" title="Link to this heading">#</a></h3>
<p>Let’s consider a well‐mixed microbial population in batch or chemostat conditions (nurients are flowing through the environment continuously), focusing on their <em>exponential growth</em> phase (AKA <em>log phase</em>). If each cell divides (on average) at rate <span class="math notranslate nohighlight">\(\lambda(T)\)</span>, then the <em>expected</em> number of cells <span class="math notranslate nohighlight">\(N(t)\)</span> follows the <em>exponential growth law</em>:</p>
<div class="math notranslate nohighlight">
\[
    \frac{dN(t)}{dt}  =  \lambda(T)  N(t),
\]</div>
<div class="math notranslate nohighlight">
\[
\implies\quad N(t)  =  N(0)  \exp\Bigl[\lambda(T)  t\Bigr].
\]</div>
<p>This is simply the standard Malthus‐type (or exponential) growth equation, whose solution is an exponential in time, with the <em>growth rate</em> <span class="math notranslate nohighlight">\(\lambda(T)\)</span> now an <em>Arrhenius‐like function</em> of temperature.</p>
<section id="growth-rate-vs-doubling-time">
<h4>Growth Rate vs Doubling Time<a class="headerlink" href="#growth-rate-vs-doubling-time" title="Link to this heading">#</a></h4>
<p>Microbiologists typically define the <em>specific growth rate</em> <span class="math notranslate nohighlight">\(\mu(T)\)</span> via</p>
<div class="math notranslate nohighlight">
\[
    \frac{dN}{dt}  =  \mu(T)  N,
\]</div>
<p>so that <span class="math notranslate nohighlight">\(\mu(T)\equiv \lambda(T)\)</span>. Equivalently, the <em>doubling time</em> is</p>
<div class="math notranslate nohighlight">
\[
    t_d  =  \frac{\ln(2)}{\mu(T)}.
\]</div>
<p>Thus, you will often see microbial growth curves written as</p>
<div class="math notranslate nohighlight">
\[
N(t)  =  N(0)  \exp\Bigl[\mu(T)  t\Bigr],
\]</div>
<p>where</p>
<div class="math notranslate nohighlight">
\[
\mu(T)  \approx  \mu_0  \exp\Bigl(-\tfrac{E_a}{R  T}\Bigr).
\]</div>
<p>So, in a nutshell,</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\( \to\)</span> Reaction rates speed up with temperature according to Arrhenius or Eyring equations.</p></li>
<li><p><span class="math notranslate nohighlight">\( \to\)</span> The <em>cell division cycle</em> (DNA replication, protein synthesis, etc.) is governed by these enzymatic rates, so the <em>cell division rate</em> <span class="math notranslate nohighlight">\(\lambda(T)\)</span> similarly increases with temperature (up to an optimum).</p></li>
<li><p><span class="math notranslate nohighlight">\( \to\)</span> When each cell divides at rate <span class="math notranslate nohighlight">\(\lambda(T)\)</span>, the <em>population</em> follows <em>exponential growth</em> <span class="math notranslate nohighlight">\(N(t) = N(0)  \exp[\lambda(T) t]\)</span>. The <em>specific growth rate</em> <span class="math notranslate nohighlight">\(\mu(T)\)</span> is effectively the <em>cell division rate</em> and inherits the same temperature dependence.</p></li>
</ul>
<p>There are also a few key caveats to keep in mind:</p>
<ul class="simple">
<li><p>Real microbes have an <em>optimal growth temperature</em>. At <em>high temperatures</em>, enzymes <em>denature</em>, and growth drops off. Thus, beyond a certain point, Arrhenius‐type behavior breaks down.</p></li>
<li><p>Many enzymatic steps contribute to growth; the overall temperature dependence can be more complicated than a simple exponential. Empirical models like the <em>Ratkowsky model</em> (<span class="math notranslate nohighlight">\(\sqrt{\mu}\)</span> vs. <span class="math notranslate nohighlight">\(T\)</span>) or the <em>Cardinal model</em> with minimum, optimum, and maximum temperatures, are often used by biologists instead of explicitly thermodynamic equations.</p></li>
<li><p>Below or near a certain threshold temperature, the <em>maintenance energy</em> requirement may exceed the energy‐generating reactions, effectively stopping growth.</p></li>
<li><p>In single‐cell studies (e.g., time‐lapse microscopy), you see that individual cells have <em>variable</em> interdivision times. However, at the <em>population level</em>, the <em>average</em> behavior typically follows the exponential law (Jafarpour et al 2019).</p></li>
</ul>
</section>
</section>
</section>
<section id="incorporating-cell-size-scaling">
<h2>Incorporating Cell Size Scaling<a class="headerlink" href="#incorporating-cell-size-scaling" title="Link to this heading">#</a></h2>
<p><em>Metabolic Theory of Ecology</em> (MTE)’s equation for organism’s <em>metabolic rate</em> (see above) can be written as:</p>
<div class="math notranslate nohighlight">
\[
    B(M, T)  \approx  B_0  M^{b}  \exp\Bigl(-\tfrac{E}{k_B  T}\Bigr),
\]</div>
<p>where:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(M\)</span> is the organism’s mass (for microbes, think <em>cell mass</em>),</p></li>
<li><p><span class="math notranslate nohighlight">\(E\)</span> is an <em>effective activation energy</em> for metabolic reactions,</p></li>
<li><p><span class="math notranslate nohighlight">\(k_B\)</span> here is Boltzmann’s constant (not to be confused with the same symbol in TST; context determines which constant is meant).</p></li>
</ul>
<p>A common assumption in MTE is that the <em>growth rate</em> of an organism is roughly <em>proportional</em> to its <em>mass‐specific</em> metabolic rate:</p>
<div class="math notranslate nohighlight">
\[
    \text{growth rate}  \propto  \frac{B(M,T)}{M}.
\]</div>
<p>Thus,</p>
<div class="math notranslate nohighlight">
\[
\text{growth rate}  \approx  \frac{B_0  M^{b}  \exp\Bigl(-E/(k_B  T)\Bigr)}{M}
     = B_0  M^{b-1}  \exp\Bigl(-E/(k_B  T)\Bigr).
\]</div>
<p>For <em>microbial cells</em> of mass <span class="math notranslate nohighlight">\(M\)</span>, the MTE‐inspired prediction for the <em>specific growth rate</em> <span class="math notranslate nohighlight">\(\mu(T, M)\)</span> becomes:</p>
<div class="math notranslate nohighlight">
\[
\mu(T, M)
 \approx 
\mu_0   M^{b-1}  \exp\Bigl(-\tfrac{E}{k_B  T}\Bigr),
\]</div>
<p>where <span class="math notranslate nohighlight">\(\mu_0\)</span> absorbs constants like <span class="math notranslate nohighlight">\(B_0\)</span>.</p>
<p>The <em><span class="math notranslate nohighlight">\(M^{b-1}\)</span></em> factor, *in the case of <em>multi-cellular organisms</em>, reflects how <em>smaller cells</em> typically have a <em>faster</em> mass‐specific metabolic rate (and thus can grow/divide faster), whereas larger cells would have a somewhat lower per‐mass metabolic rate.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The scaling of metabolic rate of microbial cells is very different (it is not allomteric, but iso-metric or super-linear; see DeLong et al 2010).</p>
</div>
<p>Next, plugging <span class="math notranslate nohighlight">\(\mu(T, M)\)</span> into the <em>exponential growth equation</em>:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}  =  \mu(T, M)  N
\quad\Longrightarrow\quad
N(t) = N(0)  \exp\Bigl[\mu(T, M)  t\Bigr].
\]</div>
<p>Hence, from MTE, the <em>population’s</em> exponential growth rate now depends explicitly on <em>both</em> Temperature (<span class="math notranslate nohighlight">\(\exp[-E/(k_B  T)]\)</span> factor and <em>organismal size</em> (<span class="math notranslate nohighlight">\(M^{1-b}\)</span> factor).</p>
<p>In simpler microbial physiology models (which ignore size variation), one might set <span class="math notranslate nohighlight">\(M\approx\text{constant}\)</span>. But if comparing cells (or species) of <em>different typical cell sizes</em>, or tracking how cell size changes under various conditions, the <span class="math notranslate nohighlight">\(M^{b-1}\)</span> dependence becomes relevant.</p>
<p>So, in summary,</p>
<ul class="simple">
<li><p>Intracellular metabolic/biochemical <em>rates</em> follow an <em>Arrhenius‐</em> or <em>Eyring‐</em> increase with <span class="math notranslate nohighlight">\(T\)</span>.</p></li>
<li><p>The <em>cell division</em> (doubling) rate <span class="math notranslate nohighlight">\(\lambda(T)\)</span> (or <span class="math notranslate nohighlight">\(\mu(T)\)</span>) is governed by these enzyme‐limited processes.</p>
<ul>
<li><p>In MTE, it also <em>scales</em> with <em>cell mass</em> as <span class="math notranslate nohighlight">\(M^{-1/4}\)</span>.</p></li>
</ul>
</li>
<li><p>When every cell divides at rate <span class="math notranslate nohighlight">\(\mu(T,M)\)</span>, the population satisfies <span class="math notranslate nohighlight">\(\tfrac{dN}{dt} = \mu(T,M)  N\)</span>. Thus, we get <em>exponential growth</em> at a <em>temperature‐ and size‐dependent</em> rate.</p></li>
</ul>
<div class="math notranslate nohighlight">
\[
\boxed{
N(t)  =  N(0)  
\exp\Bigl[\mu_0  M^{-1/4}  \exp\Bigl(-\tfrac{E}{k_B  T}\Bigr)  t\Bigr].
}
\]</div>
<p>Thus, MTE unifies the <em>thermal dependence</em> of growth rates with * Size scaling*, providing a more holistic understanding of organismal growth across different temperatures and organismal sizes across the tree of life.</p>
<section id="a-more-general-relationship">
<h3>A more general relationship<a class="headerlink" href="#a-more-general-relationship" title="Link to this heading">#</a></h3>
<p>In general, because larger multicellular organisms have lower mass-specific metabolic rates, enabling longer lifespans and greater reproductive investment (Savage et al., 2004). That is, population growth rate scales negatively with body size:</p>
<div class="math notranslate nohighlight">
\[
    r_{\text{max}} = r_0 M^{-0.25}.
\]</div>
<p>Therefore, population density decreases with size (Damuth’s law). Carnivores are rarer than herbivores due to energy loss across trophic levels (Colinvaux, 1978). We will address these phenomena later.</p>
</section>
</section>
<section id="from-exponential-to-logistic-growth">
<h2>From exponential to logistic growth<a class="headerlink" href="#from-exponential-to-logistic-growth" title="Link to this heading">#</a></h2>
<p>Next, let’s look at logistic growth, once cells and individuals have moved from the purely exponential growth regime (valid at low population density) to a density‐dependent (logistic) regime, where “carrying capacity” plays a role.</p>
<section id="classic-logistic-equation">
<h3>Classic Logistic Equation<a class="headerlink" href="#classic-logistic-equation" title="Link to this heading">#</a></h3>
<p>The <em>logistic equation</em> introduces a <em>carrying capacity</em> <span class="math notranslate nohighlight">\(K\)</span>, which is the maximum population (or density) sustainable by the environment:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt} 
 =  r  N\Bigl(1 - \frac{N}{K}\Bigr),
\]</div>
<p>where,</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(r\)</span> is the (intrinsic) per‐capita growth rate in the <em>absence</em> of density‐dependent limitation,</p></li>
<li><p><span class="math notranslate nohighlight">\(K\)</span> is the <em>carrying capacity</em>.</p></li>
</ul>
<p>If <span class="math notranslate nohighlight">\(r\)</span> and <span class="math notranslate nohighlight">\(K\)</span> are constants, the solution of this equation is well known (and which you will derive later in the course):</p>
<div class="math notranslate nohighlight">
\[ 
N(t) =  
\frac{K}{  1 + \Bigl(\tfrac{K}{N(0)} - 1\Bigr)\exp(-r  t)  },
\]</div>
<p>which saturates at <span class="math notranslate nohighlight">\(N(t)\to K\)</span> as <span class="math notranslate nohighlight">\(t\to\infty\)</span>.</p>
</section>
<section id="temperature-size-dependence-in-logistic-growth">
<h3>2.2 Temperature &amp; Size Dependence in Logistic Growth<a class="headerlink" href="#temperature-size-dependence-in-logistic-growth" title="Link to this heading">#</a></h3>
<p>To generalize, we can let the <em>intrinsic growth rate</em> <span class="math notranslate nohighlight">\(r\)</span> be <em>temperature</em>‐ and <em>cell size</em>‐dependent, i.e.,</p>
<div class="math notranslate nohighlight">
\[
r(T,M)
 \approx 
r_0  M^{-1/4}  \exp\Bigl(-\tfrac{E}{k_B  T}\Bigr).
\]</div>
<p>Thus, the logistic equation becomes:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt} = r(T,M)  N  \Bigl(1 - \frac{N}{K}\Bigr).
\]</div>
<p>Depending on the complexity of the system, one might also consider <span class="math notranslate nohighlight">\(K\)</span> to be <em>T‐dependent</em>, <span class="math notranslate nohighlight">\(M\)</span>‐dependent, or both—especially if resource supply, waste removal, or cell size changes systematically with temperature or over time.</p>
<p>When <span class="math notranslate nohighlight">\(N \ll K\)</span>, the term <span class="math notranslate nohighlight">\(\Bigl(1 - N/K\Bigr)\approx 1\)</span>, and:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}  \approx  r(T,M)  N.
\]</div>
<p>Hence, the <em>early</em> dynamics are effectively <em>exponential</em>, with the rate</p>
<div class="math notranslate nohighlight">
\[
\mu(T,M)  \approx  r(T,M) 
 \approx 
r_0  M^{-1/4}  \exp\Bigl(-\tfrac{E}{k_B  T}\Bigr).
\]</div>
<p>As <span class="math notranslate nohighlight">\(N\)</span> approaches <span class="math notranslate nohighlight">\(K\)</span>, resource depletion or waste accumulation slows growth:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}  \to  0
\quad\text{as}\quad
N  \to  K.
\]</div>
<p>Eventually, <span class="math notranslate nohighlight">\(N(t)\)</span> saturates at <span class="math notranslate nohighlight">\(K\)</span>. No matter how <em>fast</em> the intrinsic rate <span class="math notranslate nohighlight">\(r(T,M)\)</span> is (e.g., at high temperature for small cells), the population <em>cannot</em> surpass the environment’s <em>carrying capacity</em>.</p>
</section>
<section id="temperature-and-size-dependence-of-k">
<h3>Temperature and Size Dependence of <span class="math notranslate nohighlight">\(K\)</span><a class="headerlink" href="#temperature-and-size-dependence-of-k" title="Link to this heading">#</a></h3>
<p>In many real ecological or bioprocess contexts, the <em>carrying capacity</em> <span class="math notranslate nohighlight">\(K\)</span> might also <em>depend on temperature</em> (through effects on nutrient availability, oxygen solubility, etc.) or on <em>cell size</em> (larger cells might require more resources, reducing the total cell number sustainable). A simple extension might be:</p>
<div class="math notranslate nohighlight">
\[
K(T,M) =  K_0  f(T)  g(M),
\]</div>
<p>where <span class="math notranslate nohighlight">\(f(T)\)</span> and <span class="math notranslate nohighlight">\(g(M)\)</span> are functions that capture how resource utilization or environment capacity scales with temperature and cell size. Then,</p>
<div class="math notranslate nohighlight">
\[ \frac{dN}{dt} = r(T,M)  N
 \Bigl(1 - \frac{N}{K(T,M)}\Bigr).
\]</div>
<p>Although <em>analytical</em> solutions become more complicated, the qualitative behavior remains <em>logistic</em>—exponential growth at low <span class="math notranslate nohighlight">\(N\)</span>, saturating at an upper limit.</p>
</section>
</section>
<section id="population-dynamics">
<h2>Population dynamics<a class="headerlink" href="#population-dynamics" title="Link to this heading">#</a></h2>
<p>Putting all these comonents into a <em>single</em> differential equation:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}  = 
\Bigl[
  \underbrace{
    r_{0}\,M^{-\tfrac{1}{4}}\,\exp\Bigl(-\tfrac{E}{k_{B}\,T}\Bigr)
  }_{\text{Temperature \&amp; size dependent rate}}
\Bigr]
\,N(t)\,
\Bigl(1 - \tfrac{N(t)}{K}\Bigr).
\]</div>
<ul class="simple">
<li><p>When <span class="math notranslate nohighlight">\(N(t)\ll K\)</span>, <span class="math notranslate nohighlight">\(dN/dt \approx r_0  M^{-1/4}  \exp[-E/(k_B  T)]  N\)</span>.</p></li>
<li><p>As <span class="math notranslate nohighlight">\(N(t)\to K\)</span>, <span class="math notranslate nohighlight">\(dN/dt\to 0\)</span>.</p></li>
</ul>
<p>Or the more general form of <span class="math notranslate nohighlight">\(K\)</span> also depends on <span class="math notranslate nohighlight">\(T\)</span> and <span class="math notranslate nohighlight">\(M\)</span>:</p>
<div class="math notranslate nohighlight">
\[
\frac{dN}{dt}
 = 
r(T,M)  N \Bigl(1 - \frac{N}{  K(T,M)}\Bigr).
\]</div>
<p>A potentuially useful Integrated Model  is:</p>
<div class="math notranslate nohighlight">
\[ \frac{dN}{dt} = \mu(T,M) N \Bigl(1 - \frac{N}{K}\Bigr) \]</div>
<p>(optionally letting <span class="math notranslate nohighlight">\(K = K(T,M)\)</span>) succinctly unifies <em>temperature dependence</em>, <em>cell size scaling</em>, and <em>density limitations</em>.</p>
<p>Hence, you get a <em>temperature‐ and size‐dependent</em> logistic equation that captures both the <em>initial exponential ramp</em> (driven by temperature‐enhanced metabolism in smaller or larger organsims / indoviduals) and the <em>ultimate saturation</em> governed by the ecological/physical constraints of the environment.</p>
<section id="biological-interpretations">
<h3>Biological Interpretations<a class="headerlink" href="#biological-interpretations" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>Temperature and cell‐size effects dominate, so a <em>small</em> organism at an <em>optimal</em> temperature quickly ramps up in population.</p></li>
<li><p>Eventually, the environment or culture conditions impose a <em>limit</em> <span class="math notranslate nohighlight">\(K\)</span>. Growth slows down, and the population saturates (e.g., nutrient depletion in a batch culture).</p></li>
<li><p><em>Smaller</em> individuals often have <em>higher</em> mass‐specific metabolic rates, thus a <em>steeper</em> initial exponential growth. However, if <span class="math notranslate nohighlight">\(K\)</span> is in terms of total <em>number</em> of cells, smaller cells might sustain a higher <span class="math notranslate nohighlight">\(K\)</span> (because each cell needs fewer resources). Conversely, if <span class="math notranslate nohighlight">\(K\)</span> is in terms of <em>total biomass</em>, the absolute number of larger cells may be lower at carrying capacity.</p></li>
<li><p>Real systems show an <em>optimal temperature</em> for growth. Extremely high <span class="math notranslate nohighlight">\(T\)</span> leads to enzyme denaturation; extremely low <span class="math notranslate nohighlight">\(T\)</span> slows reaction rates. Thus, the functional form <span class="math notranslate nohighlight">\(r(T,M)\)</span> will be more complex than a simple Arrhenius.</p></li>
</ul>
</section>
</section>
<section id="readings">
<h2>Readings<a class="headerlink" href="#readings" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p>Arrhenius, S. (1889). <em>Über die Reaktionsgeschwindigkeit bei Inversion von Rohrzucker durch Säuren</em>. <em>Z. Phys. Chem.</em> 4, 226–248.</p></li>
<li><p>van ‘t Hoff, J. H. (1884). <em>Études de dynamique chimique</em>. Frederik Muller &amp; Cie.</p></li>
<li><p>Eyring, H. (1935). The Activated Complex in Chemical Reactions. <em>J. Chem. Phys.</em> 3(2), 107–115.</p></li>
<li><p>Polanyi, M. (1935). <em>Z. Phys. Chem. B</em> 28, 309–318.</p></li>
<li><p>Johnson, K. A., &amp; Goody, R. S. (2011). The Original Michaelis Constant: Translation of the 1913 Michaelis–Menten Paper. <em>Biochemistry</em>, 50(39), 8264–8269.</p></li>
</ol>
<!-- Below is a *non‐exhaustive list* of key references and seminal works that underlie the *theoretical* and *empirical* arguments in the derivations above—covering Arrhenius and transition state theory, microbial growth models, and the Metabolic Theory of Ecology.

---

## 1. Temperature Dependence of Reaction Rates

1. *Svante Arrhenius* (1889)  
   - *Über die Reaktionsgeschwindigkeit bei Inversion von Rohrzucker durch Säuren.*  
     *Zeitschrift für Physikalische Chemie*, *4*, 226–248.  
     - Seminal paper introducing the *Arrhenius equation* for reaction rates.

2. *Henry Eyring* (1935)  
   - *The Activated Complex in Chemical Reactions.*  
     *The Journal of Chemical Physics*, *3*, 107–115.  
     - Foundational work on *Transition State Theory (TST)*, deriving the *Eyring equation*.

3. *Keith J. Laidler* (1972)  
   - *The development of the Arrhenius equation.*  
     *Journal of Chemical Education*, *39*, 333–336.  
     - Historical perspective on how the Arrhenius equation came to be.

---

## 2. Classical Microbial Growth Kinetics

1. *F. Jacob & J. Monod* (1961)  
   - *Genetic regulatory mechanisms in the synthesis of proteins.*  
     *Journal of Molecular Biology*, *3*, 318–356.  
     - While this paper is famously about gene regulation, it lays groundwork for how enzyme kinetics tie into overall cellular processes.

2. *E. O. Powell* (1956)  
   - *Growth rate and generation time of bacteria, with special reference to continuous culture.*  
     *Journal of General Microbiology*, *15*, 492–511.  
     - Early treatments of *bacterial division rates* and how they depend on environmental conditions.

3. *M. Schaechter, O. Maaløe, & N. O. Kjeldgaard* (1958 & 1962)  
   - *Dependency on medium and temperature of cell size and chemical composition during balanced growth of Salmonella typhimurium.*  
     *Journal of General Microbiology*, *19*, 592–606 (1962).  
     - Classic experimental work linking *temperature* to *cellular physiology* (cell size, composition, growth).

4. *J. D. Murray*  
   - *Mathematical Biology* (various editions).  
     - Standard reference for *logistic growth*, *exponential growth*, and *population dynamics* modeling.

---

## 3. Logistic Growth & Population Ecology

1. *Pierre F. Verhulst* (1838)  
   - *Notice sur la loi que la population poursuit dans son accroissement.*  
     *Correspondance Mathématique et Physique*, *10*, 113–121.  
     - The original paper introducing the *logistic equation*.

2. *R. Pearl & L. J. Reed* (1920)  
   - *On the rate of growth of the population of the United States since 1790 and its mathematical representation.*  
     *Proceedings of the National Academy of Sciences (USA)*, *6*, 275–288.  
     - Classic application of logistic growth to real population data.

3. *D. A. Ratkowsky, J. Olley, T. A. McMeekin, & A. Ball* (1982)  
   - *Relationship between temperature and growth rate of bacterial cultures.*  
     *Journal of Bacteriology*, *149*, 1–5.  
     - Introduced what became the *Ratkowsky model*, an empirical expression for microbial growth rates vs. temperature, often used in food microbiology.

---

## 4. Metabolic Theory of Ecology (MTE)

1. *G. B. West, J. H. Brown, & B. J. Enquist* (1997)  
   - *A general model for the origin of allometric scaling laws in biology.*  
     *Science*, *276*, 122–126.  
     - Landmark paper deriving the *quarter‐power* scaling (e.g., $M^{3/4}$, $M^{-1/4}$) based on fractal‐like transport networks.

2. *G. B. West, J. H. Brown, & B. J. Enquist* (1999)  
   - *The Fourth Dimension of Life: Fractal Geometry and Allometric Scaling of Organisms.*  
     *Science*, *284*, 1677–1679.  
     - Follow‐up discussion on how scaling laws affect metabolism and other life processes.

3. *J. H. Brown, J. F. Gillooly, A. P. Allen, V. M. Savage, & G. B. West* (2004)  
   - *Toward a Metabolic Theory of Ecology.*  
     *Ecology*, *85*, 1771–1789.  
     - Synthesis article outlining how *metabolic rates* scale with *body mass* and *temperature*, forming the basis of *MTE*.

---

## 5. Integrative / Synthetic Works

1. *S. A. Levin* (Ed.)* (2009)  
   - *The Princeton Guide to Ecology.*  
     *Princeton University Press.*  
     - Contains chapters synthesizing *metabolic theory*, *population dynamics*, and *microbial ecology*.

2. *B. Kerr, M. A. Riley, M. W. Feldman, & B. J. M. Bohannan* (2002)  
   - *Local dispersal promotes biodiversity in a real‐life game of rock–paper–scissors.*  
     *Nature*, *418*, 171–174.  
     - While not specifically about temperature, it's an example of coupling *population dynamics* with *microbial interactions*, illustrating the complexity of real ecological systems.

3. *Metzler, D. E.* (2001)  
   - *Biochemistry: The Chemical Reactions of Living Cells.* 2nd ed., *Academic Press*.  
     - Comprehensive reference on *enzyme kinetics* in the broader physiological context.

---

### Additional Notes:

- *Empirical vs. Theoretical*: Many of the earliest derivations of *temperature‐dependent growth* arose from *empirical fits* (e.g., Arrhenius, Ratkowsky) to experimental microbial data. *Transition State Theory* was largely developed in *physical chemistry* for chemical reactions but has been adapted for *biochemical* and *enzymatic* rate processes.  
- *MTE*: Extends these ideas to *whole‐organism* (or *whole‐cell*) metabolism, adding the *allometric* component $(M^{3/4}$ scaling for metabolic rate).  
- *Logistic Growth*: Originates in *population ecology* (Verhulst, Pearl & Reed), but it's widely applicable to *microbial cultures* in a *batch* environment with finite resources.

Together, these references provide the *theoretical foundation* and *empirical evidence* for deriving and applying *temperature‐ and size‐dependent logistic (and exponential) growth* models in microbial systems. --><!-- 
## Discussion Questions

1. Given the way resting metabolic rate scales with body size, what must consumption rate scale like?
2. Why did we focus on resting (or "basal") metabolic rate?
3. How much energy would a resting rabbit (∼ 1 kg), a human (∼ 70 kg, and an Asian elephant (∼ 4000 kg) need for metabolic balance, at the individual level? What about per unit mass?
4. How does the value you calculated in the previous question compare with the recommended calorie intake for humans? Is it lower or higher? Why?
5. What are the optimal temperatures for the enzyme underlying bioluminescence in the two bacteria in the given example? Why might they have different thermal optima?
 --><p><em>Arrhenius Plots</em> are generated by taking the natural log on both sides and plotting <span class="math notranslate nohighlight">\(\ln(k)\)</span> vs. <span class="math notranslate nohighlight">\(1/T\)</span> should yield a straight line with slope <span class="math notranslate nohighlight">\(-E_a/R\)</span> and intercept <span class="math notranslate nohighlight">\(\ln(A)\)</span>.</p>
<p><em>Eyring Equation</em>:  A common linear form for plotting is:</p>
<div class="math notranslate nohighlight">
\[
   \ln\Bigl(\tfrac{k}{T}\Bigr)  =  -\frac{\Delta H^\ddagger}{R} \frac{1}{T}  +  \ln\Bigl(\frac{k_B}{h}\Bigr)  +  \frac{\Delta S^\ddagger}{R}.
\]</div>
<p>Thus, <span class="math notranslate nohighlight">\(\ln(k/T)\)</span> vs. <span class="math notranslate nohighlight">\(1/T\)</span> should be linear with slope <span class="math notranslate nohighlight">\(-\Delta H^\ddagger / R\)</span> and intercept <span class="math notranslate nohighlight">\(\ln(k_B/h) + \Delta S^\ddagger / R\)</span>.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notebooks"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="R.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Biological Computing in R</p>
      </div>
    </a>
    <a class="right-next"
       href="Populations.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Single Populations</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#introduction">Introduction</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#energy-and-metabolic-rate">Energy and Metabolic Rate</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#importance-of-size">Importance of Size</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#geometric-foundations-of-the-metabolic-scaling-law">Geometric foundations of the metabolic scaling law</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#biological-implications-of-the-geometric-scaling-constraint">Biological implications of the geometric scaling constraint</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#going-beyond-geometric-constraints">Going beyond geometric constraints</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#importance-of-temperature-and-thermodynamic-constraints">Importance of Temperature (and Thermodynamic constraints)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#classical-transition-state-theory-tst">Classical Transition State Theory (TST)</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overall-temperature-dependence-of-the-rate">Overall Temperature Dependence of the Rate</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#putting-it-together-temperature-size-scaling-and-metabolic-rate">Putting it together: temperature, size scaling, and metabolic rate</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#ecological-implications">Ecological implications</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#from-intracellular-reaction-rates-to-cell-growth">From intracellular reaction rates to cell growth</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#from-singlecell-growth-to-division">From single‐cell growth to division</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#cell-division-is-a-stochastic-process">Cell division is a stochastic process</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#average-doubling-time">(Average) Doubling time</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#from-cell-division-rate-to-population-growth">From cell division rate to population growth</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#growth-rate-vs-doubling-time">Growth Rate vs Doubling Time</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#incorporating-cell-size-scaling">Incorporating Cell Size Scaling</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#a-more-general-relationship">A more general relationship</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#from-exponential-to-logistic-growth">From exponential to logistic growth</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#classic-logistic-equation">Classic Logistic Equation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#temperature-size-dependence-in-logistic-growth">2.2 Temperature &amp; Size Dependence in Logistic Growth</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#temperature-and-size-dependence-of-k">Temperature and Size Dependence of <span class="math notranslate nohighlight">\(K\)</span></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#population-dynamics">Population dynamics</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#biological-interpretations">Biological Interpretations</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#readings">Readings</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By MulQuaBio collective!
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>